{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import chi2\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "import urllib\n",
    "\n",
    "# Read a .txt or .csv file containing SNPs and P values to be plotted\n",
    "# Function checks delimiters automatically\n",
    "# Parameter: filename - .txt or .csv file containing information needed\n",
    "# Return: pandas dataframe of the file\n",
    "def __read_file(filename):\n",
    "    fh = open(filename, 'r')\n",
    "    line = fh.readline()\n",
    "    tmp = line.strip().split(',')\n",
    "\n",
    "    # Create an empty dataframe, read data into df later\n",
    "    df = pd.DataFrame()\n",
    "    # Check delimiter type, \",\" or \" \"\n",
    "    if len(tmp) == 1:\n",
    "        df = pd.read_csv(filename, delim_whitespace=True)\n",
    "    else:\n",
    "        df = pd.read_csv(filename)\n",
    "    \n",
    "    return(df)\n",
    "\n",
    "# Return -log10 transformed observed and expected p vlaues of a dataset(dataframe)\n",
    "def __log_obsv_and_expt(df, column_title):\n",
    "    Pobsv = df.loc[:, column_title]  # Observed p values\n",
    "    Pobsv = np.sort(Pobsv.dropna().values)\n",
    "    Pexpt = np.arange(1, len(Pobsv) + 1, 1) / (len(Pobsv) + 1)  # Expected\n",
    "    logPobsv = -np.log10(Pobsv)\n",
    "    logPexpt = -np.log10(Pexpt)\n",
    "    \n",
    "    return(Pobsv, logPobsv, logPexpt)\n",
    "\n",
    "\n",
    "# Calculate and return lambda (inflation)\n",
    "def __get_inflation__(observed):\n",
    "    obsv_median = np.median(observed)\n",
    "    Chi = chi2.ppf(1.0 - obsv_median, 1)\n",
    "    lmbd = Chi / chi2.ppf(0.5, 1)\n",
    "    return lmbd\n",
    "\n",
    "\n",
    "# The tricky part for me is to query haploreg database thorugh their website.\n",
    "# Inspecct the element of the website to find out names of each variable\n",
    "# Recommand using Chrome, go to developer mode, refresh the website under Network tab\n",
    "# This function is intended to query one SNP at a time\n",
    "# Return: a set of rsIDs of accociated SNPs (and the queried SNP)\n",
    "\n",
    "# Parameters needed to send to haploreg website, and some explaination\n",
    "# input_snps (query): input SNPs in a string, each SNP is separated by ','\n",
    "#        SNP number per query is set to be 1000, otherwise it takes too long and haploreg may refuse to do it.\n",
    "#        15000 SNPs did not work for test run.\n",
    "# gwas_id: the dropdown list to choose GWAS study, when no file or query SNP(s) is provided\n",
    "# r2_threshold (ldThresh): r^2 threshold, default is 0.2 in this code\n",
    "# ldPop: 1000G Phase 1 population for LD calculation\n",
    "# epi: Source for epigenomes\n",
    "# cons: Mammalian conservation algorithm. 'siphy'=SiPhy-omega, 'gerp'=GERP, or 'both'\n",
    "# genetypes: Show position relative to\n",
    "# output: set output result type to 'text' for python code to process\n",
    "def single_query_haploreg(input_snp,\n",
    "                   r2_threshold=0.2,\n",
    "                   ldPop='EUR',\n",
    "                   epi='vanilla',\n",
    "                   cons='siphy',\n",
    "                   genetypes='gencode'):\n",
    "\n",
    "    params_library = {'query':input_snp,\n",
    "                      'gwas_id':0,\n",
    "                      'ldThresh': r2_threshold,\n",
    "                      'ldPop': ldPop,\n",
    "                      'epi': epi,\n",
    "                      'cons': cons,\n",
    "                      'genetypes': genetypes,\n",
    "                      'output':'text'}\n",
    "    \n",
    "    # parameters passed to the website, needs to be parsed and in binary\n",
    "    params = urllib.parse.urlencode(params_library).encode(\"utf-8\")\n",
    "    # url of HaploReg4.1\n",
    "    url = 'https://pubs.broadinstitute.org/mammals/haploreg/haploreg.php'\n",
    "    # Query with parameters\n",
    "    query = urllib.request.urlopen(url, params)\n",
    "    content = query.read().decode(\"utf-8\")\n",
    "    # Find accociated SNPs in the content returned from HaploReg\n",
    "    matches = re.findall('rs[0-9]+', content)\n",
    "\n",
    "    # Return unique ones of the query\n",
    "    return(set(matches))\n",
    "#     return content\n",
    "\n",
    "\n",
    "# Queried SNPs are based on the file provided\n",
    "# Parameter:\n",
    "# - \n",
    "# -\n",
    "# Return: Save novel SNPs as .csv file fot plotting and reference\n",
    "def __novel_SNPs(filename, r2_threshold, SNPs_column_title, novel_filename='novel_SNPs.csv'):\n",
    "    original_df = __read_file(filename) # Dataframe read from the file\n",
    "    all_SNPs = original_df.loc[:, SNPs_column_title].values # Get all SNPs in the file to be queried\n",
    "    associated_SNPs = np.array([]) # A numpy array to store queried results\n",
    "    \n",
    "    # Count is to show numbers of SNPs processed so far, to keep the console updating\n",
    "    count = 0\n",
    "    for snp in all_SNPs:\n",
    "        associated_SNPs = np.append(associated_SNPs, single_query_haploreg(input_snp=snp, r2_threshold=r2_threshold))\n",
    "        \n",
    "        count = count+1\n",
    "        # Print every 10 SNPs\n",
    "        if count%10 == 0:\n",
    "            print('Number of SNPs processed: ' + str(count))\n",
    "        \n",
    "    \n",
    "    # Remove duplicates of queried results\n",
    "    associated_SNPs = np.unique(associated_SNPs)\n",
    "    \n",
    "    # Join the original dataframe and queried results, output novel SNPs\n",
    "    original_df.set_index(SNPs_column_title, inplace=True)\n",
    "    associated_df = pd.DataFrame([], index=associated_SNPs)\n",
    "    remove_df = original_df.join(associated_df, how='inner')\n",
    "    novel_df = original_df.drop(remove_df.index.values)\n",
    "    \n",
    "    novel_df.to_csv(novel_filename)\n",
    "\n",
    "# Returns: (fig, ax, lambda)\n",
    "# fig and ax for more custermizations\n",
    "def qqplot(filename,\n",
    "           output='output.png',\n",
    "           column_title = 'P',\n",
    "           title='Q-Q plot',\n",
    "           xlabel='Expected –log10 P-values',\n",
    "           ylabel='Observed –log10 P-values',\n",
    "           dpi=300,\n",
    "           plot_novel = False,\n",
    "           novel_filename = 'novel_SNPs.csv',\n",
    "           SNPs_column_title = 'SNP',\n",
    "           r2_threshold=0.2):\n",
    "    \n",
    "    df = __read_file(filename)\n",
    "    \n",
    "#     Pobsv = df.loc[:, column_title]  # Observed p values\n",
    "#     Pobsv = np.sort(Pobsv.dropna().values)\n",
    "#     Pexpt = np.arange(1, len(Pobsv) + 1, 1) / (len(Pobsv) + 1)  # Expected\n",
    "#     logPobsv = -np.log10(Pobsv)\n",
    "#     logPexpt = -np.log10(Pexpt)\n",
    "\n",
    "    Pobsv, logPobsv, logPexpt = __log_obsv_and_expt(df, column_title)\n",
    "\n",
    "    # Calculate lambda\n",
    "    infl = __get_inflation__(Pobsv)\n",
    "\n",
    "    fig, ax = plt.subplots(dpi=dpi)\n",
    "    ax.plot(logPexpt, logPobsv, linestyle='', marker='o', markersize=3, markeredgewidth=0.5,\n",
    "            fillstyle='none', color='k', alpha=0.8)\n",
    "    ax.plot(logPexpt, logPexpt, color='r', linewidth=0.4)\n",
    "\n",
    "    # Label x and y axis\n",
    "    ax.set_title(title)\n",
    "    ax.set_xlabel(xlabel)\n",
    "    ax.set_ylabel(ylabel)\n",
    "    annotation = \"λ = \" + str(\"{0:.4f}\".format(infl))\n",
    "    ax.annotate(annotation, xy=(0.7, 0.2), xycoords='axes fraction')\n",
    "    \n",
    "    if plot_novel == True:\n",
    "        # Plot novel SNPs, then same steps as before\n",
    "        __novel_SNPs(filename, r2_threshold, SNPs_column_title, novel_filename)\n",
    "        novel_SNPs_df = __read_file(novel_filename)\n",
    "        \n",
    "#         Pobsv_novel = novel_SNPs_df.loc[:, column_title]  # Observed p values\n",
    "#         Pobsv_novel = np.sort(Pobsv_novel.dropna().values)\n",
    "#         Pexpt_novel = np.arange(1, len(Pobsv_novel) + 1, 1) / (len(Pobsv_novel) + 1)  # Expected\n",
    "#         logPobsv_novel = -np.log10(Pobsv_novel)\n",
    "#         logPexpt_novel = -np.log10(Pexpt_novel)\n",
    "        Pobsv_novel, logPobsv_novel, logPexpt_novel = __log_obsv_and_expt(novel_SNPs_df, column_title)\n",
    "\n",
    "        ax.plot(logPexpt_novel, logPobsv_novel, linestyle='', marker='o', markersize=3, markeredgewidth=0.5,\n",
    "                fillstyle='none', color='g', alpha=0.8)\n",
    "    \n",
    "    fig.savefig(output)\n",
    "    \n",
    "    return (fig, ax, infl)  # Return for more custermizations\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Takes a df and scan for novel SNPs\n",
    "# Better to process about 100 SNPs at a time, so the process will not take too long.\n",
    "# column_title is the titile of the column that contains rsIDs\n",
    "# Output the reuslt into a .csv file\n",
    "def __associated_SNPs(file_df, column_title='SNP', r2_threshold=0.2, output='associated_SNPs.txt'):\n",
    "    all_SNPs = file_df.loc[:,column_title].values # Get all SNPs in the file to be queried\n",
    "    associated_SNPs = set() # A numpy array to store queried results\n",
    "\n",
    "    for snp in all_SNPs:\n",
    "        associated_SNPs = np.append(associated_SNPs, single_query_haploreg(input_snp=snp, r2_threshold=r2_threshold))\n",
    "    \n",
    "    # Remove duplicates of queried results\n",
    "    associated_SNPs = np.unique(associated_SNPs)\n",
    "    np.savetxt(output, associated_SNPs, fmt='%s')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# This can be single or multiple SNP queris (each rsID is separated by ',')\n",
    "def query_haploreg(input_snp,\n",
    "                   r2_threshold=0.2,\n",
    "                   ldPop='EUR',\n",
    "                   epi='vanilla',\n",
    "                   cons='siphy',\n",
    "                   genetypes='gencode'):\n",
    "\n",
    "    params_library = {'query':input_snp,\n",
    "                      'gwas_id':0,\n",
    "                      'ldThresh': r2_threshold,\n",
    "                      'ldPop': ldPop,\n",
    "                      'epi': epi,\n",
    "                      'cons': cons,\n",
    "                      'genetypes': genetypes,\n",
    "                      'output':'text'}\n",
    "    \n",
    "    # parameters passed to the website, needs to be parsed and in binary\n",
    "    params = urllib.parse.urlencode(params_library).encode(\"utf-8\")\n",
    "    # url of HaploReg4.1\n",
    "    url = 'https://pubs.broadinstitute.org/mammals/haploreg/haploreg.php'\n",
    "    # Query with parameters\n",
    "    query = urllib.request.urlopen(url, params)\n",
    "    content = query.read().decode(\"utf-8\")\n",
    "    # Find accociated SNPs in the content returned from HaploReg\n",
    "    matches = re.findall('rs[0-9]+', content)\n",
    "\n",
    "    # Return unique ones of the query\n",
    "    return(list(set(matches)))\n",
    "\n",
    "# This function returns known SNPs (and associated SNPs) based on user input file\n",
    "# Parameters:\n",
    "#  - filename: a file contains known SNPs, provided by user\n",
    "#  - r2_threshold: LD threshold, defualt is 0.2\n",
    "#  - SNPs_column_title: Column title of SNPs in that file\n",
    "# Return: a list of associated SNPs pulled from HaploReg and known SNPs provided by user\n",
    "def __known_SNPs(filename, r2_threshold=0.2, SNPs_column_title='SNP'):\n",
    "    known_SNPs_df = __read_file(filename) # Dataframe of known SNPs\n",
    "    # Get all SNPs to be queried, join them by ',' and query together\n",
    "    known_SNPs = ','.join(known_SNPs_df.loc[:, SNPs_column_title].values)\n",
    "    # A numpy array to store queried results\n",
    "    associated_and_known_SNPs = query_haploreg(input_snp=known_SNPs, r2_threshold=r2_threshold)\n",
    "    \n",
    "#     if len(known_SNPs_df)>100:\n",
    "\n",
    "\n",
    "    # Remove duplicates of queried results and return\n",
    "    return(set(associated_and_known_SNPs))\n",
    "\n",
    "# This function removes known SNPs and save novel SNPs in a .csv file\n",
    "# Parameters:\n",
    "#  - original_df: a dataframe containing raw data of SNPs and p values\n",
    "#  - known_SNPs: a list of SNPs found in other GWAS, that are associated with traits of interest\n",
    "#  - SNPs_column_title: Column title of SNPs in that file\n",
    "# Return: a datafram of novel SNPs and p values\n",
    "def __remove_known_SNPs(original_df, known_SNPs, SNPs_column_title='SNP', novel_filename='novel_SNPs'):\n",
    "    # Join the original dataframe and konw SNPs, output novel SNPs to be plotted\n",
    "    original_df.set_index(SNPs_column_title, inplace=True)\n",
    "    known_SNPs_df = pd.DataFrame([], index=known_SNPs)\n",
    "    to_be_removed_SNPs_df = original_df.join(known_SNPs_df, how='inner')\n",
    "    novel_df = original_df.drop(to_be_removed_SNPs_df.index.values)\n",
    "    \n",
    "    novel_df.to_csv(novel_filename)\n",
    "    return(novel_df)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "filename = 'Data/DLD_GWAS_10.csv'\n",
    "a = __known_SNPs(filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "original_df = __read_file('Data/ADD_only_DLD_GWAS.txt')\n",
    "new_df = __remove_known_SNPs(original_df, a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "gwas_filename_01 = \"/data100t1/share/DIAMANTE/mrmega_hdl_final_0619_rsids_locuszoom.result\"\n",
    "gwas_filename_02 = \"/data100t1/share/DIAMANTE/mrmega_hdl_final_0619_rsids.result\"\n",
    "\n",
    "gwas_df = pd.read_csv(gwas_filename_02, sep='\\t')\n",
    "\n",
    "# fh = open(gwas_filename_02, 'r')\n",
    "# count=0\n",
    "# for line in fh:\n",
    "#     print(line.strip().split('\\t'))\n",
    "#     count=count+1\n",
    "#     if count>10: break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MarkerName</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>EA</th>\n",
       "      <th>NEA</th>\n",
       "      <th>EAF</th>\n",
       "      <th>Nsample</th>\n",
       "      <th>Ncohort</th>\n",
       "      <th>Effects</th>\n",
       "      <th>beta_0</th>\n",
       "      <th>...</th>\n",
       "      <th>ndf_association</th>\n",
       "      <th>P-value_association</th>\n",
       "      <th>chisq_ancestry_het</th>\n",
       "      <th>ndf_ancestry_het</th>\n",
       "      <th>P-value_ancestry_het</th>\n",
       "      <th>chisq_residual_het</th>\n",
       "      <th>ndf_residual_het</th>\n",
       "      <th>P-value_residual_het</th>\n",
       "      <th>lnBF</th>\n",
       "      <th>Comments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs147324274</td>\n",
       "      <td>10</td>\n",
       "      <td>100000012</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.996606</td>\n",
       "      <td>26037</td>\n",
       "      <td>4</td>\n",
       "      <td>-?????????+??????++</td>\n",
       "      <td>-1.252750</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.317464</td>\n",
       "      <td>0.677718</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.410374</td>\n",
       "      <td>0.223172</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.894415</td>\n",
       "      <td>-0.238903</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs144804129</td>\n",
       "      <td>10</td>\n",
       "      <td>100000122</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.997312</td>\n",
       "      <td>27449</td>\n",
       "      <td>6</td>\n",
       "      <td>-+?????+??+??????+-</td>\n",
       "      <td>0.086105</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.765692</td>\n",
       "      <td>0.000672</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.979314</td>\n",
       "      <td>3.196730</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.525459</td>\n",
       "      <td>-1.524780</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs6602381</td>\n",
       "      <td>10</td>\n",
       "      <td>10000018</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.541823</td>\n",
       "      <td>38176</td>\n",
       "      <td>19</td>\n",
       "      <td>+++--+-++-+-+--++++</td>\n",
       "      <td>0.029652</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.295628</td>\n",
       "      <td>0.875949</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.349314</td>\n",
       "      <td>12.777500</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.750949</td>\n",
       "      <td>-1.725790</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs189891329</td>\n",
       "      <td>10</td>\n",
       "      <td>10000033</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.994151</td>\n",
       "      <td>26037</td>\n",
       "      <td>4</td>\n",
       "      <td>-?????????+??????+-</td>\n",
       "      <td>-1.370290</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.475657</td>\n",
       "      <td>1.028940</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.310409</td>\n",
       "      <td>7.119850</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.028441</td>\n",
       "      <td>-0.643236</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs112832083</td>\n",
       "      <td>10</td>\n",
       "      <td>100000588</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.992032</td>\n",
       "      <td>28518</td>\n",
       "      <td>7</td>\n",
       "      <td>+??-?+?+??+??????+-</td>\n",
       "      <td>-0.237271</td>\n",
       "      <td>...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.093044</td>\n",
       "      <td>1.776260</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.182610</td>\n",
       "      <td>2.751990</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.738157</td>\n",
       "      <td>0.428771</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    MarkerName  Chromosome   Position EA NEA       EAF  Nsample  Ncohort  \\\n",
       "0  rs147324274          10  100000012  G   A  0.996606    26037        4   \n",
       "1  rs144804129          10  100000122  T   A  0.997312    27449        6   \n",
       "2    rs6602381          10   10000018  A   G  0.541823    38176       19   \n",
       "3  rs189891329          10   10000033  G   A  0.994151    26037        4   \n",
       "4  rs112832083          10  100000588  T   C  0.992032    28518        7   \n",
       "\n",
       "               Effects    beta_0  ...  ndf_association  P-value_association  \\\n",
       "0  -?????????+??????++ -1.252750  ...              2.0             0.317464   \n",
       "1  -+?????+??+??????+-  0.086105  ...              2.0             0.765692   \n",
       "2  +++--+-++-+-+--++++  0.029652  ...              2.0             0.295628   \n",
       "3  -?????????+??????+- -1.370290  ...              2.0             0.475657   \n",
       "4  +??-?+?+??+??????+- -0.237271  ...              2.0             0.093044   \n",
       "\n",
       "   chisq_ancestry_het  ndf_ancestry_het  P-value_ancestry_het  \\\n",
       "0            0.677718               1.0              0.410374   \n",
       "1            0.000672               1.0              0.979314   \n",
       "2            0.875949               1.0              0.349314   \n",
       "3            1.028940               1.0              0.310409   \n",
       "4            1.776260               1.0              0.182610   \n",
       "\n",
       "   chisq_residual_het  ndf_residual_het  P-value_residual_het      lnBF  \\\n",
       "0            0.223172               2.0              0.894415 -0.238903   \n",
       "1            3.196730               4.0              0.525459 -1.524780   \n",
       "2           12.777500              17.0              0.750949 -1.725790   \n",
       "3            7.119850               2.0              0.028441 -0.643236   \n",
       "4            2.751990               5.0              0.738157  0.428771   \n",
       "\n",
       "   Comments  \n",
       "0       NaN  \n",
       "1       NaN  \n",
       "2       NaN  \n",
       "3       NaN  \n",
       "4       NaN  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "33512268"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gwas_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "33783872"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gwas_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(<Figure size 1800x1200 with 1 Axes>,\n",
       " <matplotlib.axes._subplots.AxesSubplot at 0x7fc6383c5cf8>,\n",
       " 1.0720526881638353)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "qqplot(filename=gwas_filename_01,output='20190930_mrmega_hdl.png',\n",
    "       column_title = 'P-value_association',title='mrmega_hdl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "novel_01 = '/data100t1/share/DIAMANTE/GWAS_catalog_results_HDL.txt'\n",
    "novel_df = pd.read_csv(novel_01, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DATE ADDED TO CATALOG</th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>FIRST AUTHOR</th>\n",
       "      <th>DATE</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>INITIAL SAMPLE SIZE</th>\n",
       "      <th>REPLICATION SAMPLE SIZE</th>\n",
       "      <th>...</th>\n",
       "      <th>PVALUE_MLOG</th>\n",
       "      <th>P-VALUE (TEXT)</th>\n",
       "      <th>OR or BETA</th>\n",
       "      <th>95% CI (TEXT)</th>\n",
       "      <th>PLATFORM [SNPS PASSING QC]</th>\n",
       "      <th>CNV</th>\n",
       "      <th>MAPPED_TRAIT</th>\n",
       "      <th>MAPPED_TRAIT_URI</th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <th>GENOTYPING TECHNOLOGY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6/21/17</td>\n",
       "      <td>28270201</td>\n",
       "      <td>Nagy R</td>\n",
       "      <td>3/7/17</td>\n",
       "      <td>Genome Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28270201</td>\n",
       "      <td>Exploration of haplotype research consortium i...</td>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>19,223 British ancestry individuals from 6863 ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>7.522879</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.867736</td>\n",
       "      <td>[0.56-1.18] mg/dl increase</td>\n",
       "      <td>Illumina [24111857] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST004207</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2/14/19</td>\n",
       "      <td>29507422</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>13.698970</td>\n",
       "      <td>(EA)</td>\n",
       "      <td>0.113000</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2/14/19</td>\n",
       "      <td>29507422</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>13.698970</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.114000</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5/12/14</td>\n",
       "      <td>24097068</td>\n",
       "      <td>Willer CJ</td>\n",
       "      <td>10/6/13</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/24097068</td>\n",
       "      <td>Discovery and refinement of loci associated wi...</td>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>94,595 European ancestry individuals</td>\n",
       "      <td>93,982 European ancestry individuals</td>\n",
       "      <td>...</td>\n",
       "      <td>15.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.051000</td>\n",
       "      <td>[NR] unit decrease</td>\n",
       "      <td>NR [NR] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST002223</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6/26/17</td>\n",
       "      <td>28334899</td>\n",
       "      <td>Spracklen CN</td>\n",
       "      <td>2/21/17</td>\n",
       "      <td>Hum Mol Genet</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28334899</td>\n",
       "      <td>Association analyses of East Asian individuals...</td>\n",
       "      <td>HDL cholesterol levels</td>\n",
       "      <td>34,930 East Asian ancestry individuals, 187,16...</td>\n",
       "      <td>8,741 Chinese ancestry individuals</td>\n",
       "      <td>...</td>\n",
       "      <td>15.045757</td>\n",
       "      <td>(Trans-ethnic initial)</td>\n",
       "      <td>0.050600</td>\n",
       "      <td>[0.038-0.063] unit decrease (EA Beta values)</td>\n",
       "      <td>Affymetrix, Illumina [~ 1900000] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST004232</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG  PUBMEDID  FIRST AUTHOR     DATE        JOURNAL  \\\n",
       "0               6/21/17  28270201        Nagy R   3/7/17     Genome Med   \n",
       "1               2/14/19  29507422   Hoffmann TJ   3/5/18      Nat Genet   \n",
       "2               2/14/19  29507422   Hoffmann TJ   3/5/18      Nat Genet   \n",
       "3               5/12/14  24097068     Willer CJ  10/6/13      Nat Genet   \n",
       "4               6/26/17  28334899  Spracklen CN  2/21/17  Hum Mol Genet   \n",
       "\n",
       "                                   LINK  \\\n",
       "0  www.ncbi.nlm.nih.gov/pubmed/28270201   \n",
       "1  www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "2  www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "3  www.ncbi.nlm.nih.gov/pubmed/24097068   \n",
       "4  www.ncbi.nlm.nih.gov/pubmed/28334899   \n",
       "\n",
       "                                               STUDY  \\\n",
       "0  Exploration of haplotype research consortium i...   \n",
       "1  A large electronic-health-record-based genome-...   \n",
       "2  A large electronic-health-record-based genome-...   \n",
       "3  Discovery and refinement of loci associated wi...   \n",
       "4  Association analyses of East Asian individuals...   \n",
       "\n",
       "                                 DISEASE/TRAIT  \\\n",
       "0                              HDL cholesterol   \n",
       "1  High density lipoprotein cholesterol levels   \n",
       "2  High density lipoprotein cholesterol levels   \n",
       "3                              HDL cholesterol   \n",
       "4                       HDL cholesterol levels   \n",
       "\n",
       "                                 INITIAL SAMPLE SIZE  \\\n",
       "0  19,223 British ancestry individuals from 6863 ...   \n",
       "1  76,627 European ancestry individuals, 7,795 Hi...   \n",
       "2  76,627 European ancestry individuals, 7,795 Hi...   \n",
       "3               94,595 European ancestry individuals   \n",
       "4  34,930 East Asian ancestry individuals, 187,16...   \n",
       "\n",
       "                REPLICATION SAMPLE SIZE  ... PVALUE_MLOG  \\\n",
       "0                                   NaN  ...    7.522879   \n",
       "1                                   NaN  ...   13.698970   \n",
       "2                                   NaN  ...   13.698970   \n",
       "3  93,982 European ancestry individuals  ...   15.000000   \n",
       "4    8,741 Chinese ancestry individuals  ...   15.045757   \n",
       "\n",
       "           P-VALUE (TEXT) OR or BETA  \\\n",
       "0                     NaN   0.867736   \n",
       "1                    (EA)   0.113000   \n",
       "2                     NaN   0.114000   \n",
       "3                     NaN   0.051000   \n",
       "4  (Trans-ethnic initial)   0.050600   \n",
       "\n",
       "                                  95% CI (TEXT)  \\\n",
       "0                    [0.56-1.18] mg/dl increase   \n",
       "1                                 unit increase   \n",
       "2                                 unit increase   \n",
       "3                            [NR] unit decrease   \n",
       "4  [0.038-0.063] unit decrease (EA Beta values)   \n",
       "\n",
       "                   PLATFORM [SNPS PASSING QC] CNV  \\\n",
       "0               Illumina [24111857] (imputed)   N   \n",
       "1     Affymetrix [at least 7091467] (imputed)   N   \n",
       "2     Affymetrix [at least 7091467] (imputed)   N   \n",
       "3                           NR [NR] (imputed)   N   \n",
       "4  Affymetrix, Illumina [~ 1900000] (imputed)   N   \n",
       "\n",
       "                                       MAPPED_TRAIT  \\\n",
       "0  high density lipoprotein cholesterol measurement   \n",
       "1  high density lipoprotein cholesterol measurement   \n",
       "2  high density lipoprotein cholesterol measurement   \n",
       "3  high density lipoprotein cholesterol measurement   \n",
       "4  high density lipoprotein cholesterol measurement   \n",
       "\n",
       "                       MAPPED_TRAIT_URI  STUDY ACCESSION  \\\n",
       "0  http://www.ebi.ac.uk/efo/EFO_0004612       GCST004207   \n",
       "1  http://www.ebi.ac.uk/efo/EFO_0004612       GCST007140   \n",
       "2  http://www.ebi.ac.uk/efo/EFO_0004612       GCST007140   \n",
       "3  http://www.ebi.ac.uk/efo/EFO_0004612       GCST002223   \n",
       "4  http://www.ebi.ac.uk/efo/EFO_0004612       GCST004232   \n",
       "\n",
       "          GENOTYPING TECHNOLOGY  \n",
       "0  Genome-wide genotyping array  \n",
       "1  Genome-wide genotyping array  \n",
       "2  Genome-wide genotyping array  \n",
       "3  Genome-wide genotyping array  \n",
       "4  Genome-wide genotyping array  \n",
       "\n",
       "[5 rows x 38 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "novel_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "38"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(novel_df.T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>1290</th>\n",
       "      <th>1291</th>\n",
       "      <th>1292</th>\n",
       "      <th>1293</th>\n",
       "      <th>1294</th>\n",
       "      <th>1295</th>\n",
       "      <th>1296</th>\n",
       "      <th>1297</th>\n",
       "      <th>1298</th>\n",
       "      <th>1299</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DATE ADDED TO CATALOG</th>\n",
       "      <td>6/21/17</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>5/12/14</td>\n",
       "      <td>6/26/17</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>11/26/18</td>\n",
       "      <td>...</td>\n",
       "      <td>7/28/18</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>11/26/18</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>11/26/18</td>\n",
       "      <td>11/26/18</td>\n",
       "      <td>2/14/19</td>\n",
       "      <td>2/14/19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PUBMEDID</th>\n",
       "      <td>28270201</td>\n",
       "      <td>29507422</td>\n",
       "      <td>29507422</td>\n",
       "      <td>24097068</td>\n",
       "      <td>28334899</td>\n",
       "      <td>29507422</td>\n",
       "      <td>29507422</td>\n",
       "      <td>29507422</td>\n",
       "      <td>29507422</td>\n",
       "      <td>30275531</td>\n",
       "      <td>...</td>\n",
       "      <td>29403010</td>\n",
       "      <td>29507422</td>\n",
       "      <td>29507422</td>\n",
       "      <td>30275531</td>\n",
       "      <td>29507422</td>\n",
       "      <td>29507422</td>\n",
       "      <td>30275531</td>\n",
       "      <td>30275531</td>\n",
       "      <td>29507422</td>\n",
       "      <td>29507422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FIRST AUTHOR</th>\n",
       "      <td>Nagy R</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Willer CJ</td>\n",
       "      <td>Spracklen CN</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Klarin D</td>\n",
       "      <td>...</td>\n",
       "      <td>Kanai M</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Klarin D</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Klarin D</td>\n",
       "      <td>Klarin D</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DATE</th>\n",
       "      <td>3/7/17</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>10/6/13</td>\n",
       "      <td>2/21/17</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>10/1/18</td>\n",
       "      <td>...</td>\n",
       "      <td>2/5/18</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>10/1/18</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>10/1/18</td>\n",
       "      <td>10/1/18</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>3/5/18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>JOURNAL</th>\n",
       "      <td>Genome Med</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Hum Mol Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>...</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>Nat Genet</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LINK</th>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28270201</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/24097068</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28334899</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/30275531</td>\n",
       "      <td>...</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29403010</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/30275531</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/30275531</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/30275531</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STUDY</th>\n",
       "      <td>Exploration of haplotype research consortium i...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>Discovery and refinement of loci associated wi...</td>\n",
       "      <td>Association analyses of East Asian individuals...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>Genetics of blood lipids among ~300,000 multi-...</td>\n",
       "      <td>...</td>\n",
       "      <td>Genetic analysis of quantitative traits in the...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>Genetics of blood lipids among ~300,000 multi-...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>Genetics of blood lipids among ~300,000 multi-...</td>\n",
       "      <td>Genetics of blood lipids among ~300,000 multi-...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>HDL cholesterol levels</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>...</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INITIAL SAMPLE SIZE</th>\n",
       "      <td>19,223 British ancestry individuals from 6863 ...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>94,595 European ancestry individuals</td>\n",
       "      <td>34,930 East Asian ancestry individuals, 187,16...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>215,551 European ancestry individuals, 57,332 ...</td>\n",
       "      <td>...</td>\n",
       "      <td>70,657 Japanese ancestry individuals</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>215,551 European ancestry individuals, 57,332 ...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>215,551 European ancestry individuals, 57,332 ...</td>\n",
       "      <td>215,551 European ancestry individuals, 57,332 ...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REPLICATION SAMPLE SIZE</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>93,982 European ancestry individuals</td>\n",
       "      <td>8,741 Chinese ancestry individuals</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Up to 319,677 individuals</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Up to 319,677 individuals</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Up to 319,677 individuals</td>\n",
       "      <td>Up to 319,677 individuals</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REGION</th>\n",
       "      <td>1p36.23</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>1p36.11</td>\n",
       "      <td>...</td>\n",
       "      <td>22q11.21</td>\n",
       "      <td>22q11.21</td>\n",
       "      <td>22q11.21</td>\n",
       "      <td>22q12.1</td>\n",
       "      <td>22q13.1</td>\n",
       "      <td>22q13.1</td>\n",
       "      <td>22q13.1</td>\n",
       "      <td>22q13.31</td>\n",
       "      <td>Xp22.2</td>\n",
       "      <td>Xp22.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHR_ID</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>X</td>\n",
       "      <td>X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHR_POS</th>\n",
       "      <td>8,173,210</td>\n",
       "      <td>26,661,155</td>\n",
       "      <td>26,661,155</td>\n",
       "      <td>26,811,902</td>\n",
       "      <td>26,811,902</td>\n",
       "      <td>26,811,902</td>\n",
       "      <td>26,811,902</td>\n",
       "      <td>26,958,422</td>\n",
       "      <td>26,958,422</td>\n",
       "      <td>26,958,422</td>\n",
       "      <td>...</td>\n",
       "      <td>21,622,758</td>\n",
       "      <td>21,627,765</td>\n",
       "      <td>21,627,765</td>\n",
       "      <td>29,004,527</td>\n",
       "      <td>38,179,492</td>\n",
       "      <td>38,179,492</td>\n",
       "      <td>38,198,661</td>\n",
       "      <td>43,945,024</td>\n",
       "      <td>14,841,981</td>\n",
       "      <td>14,841,981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>REPORTED GENE(S)</th>\n",
       "      <td>intergenic</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>PIGV, NR0B2</td>\n",
       "      <td>PIGV, NROB2</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>C1orf172</td>\n",
       "      <td>...</td>\n",
       "      <td>UBE2L3</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>MAFF</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <td>AL034417.2 - AL358876.1</td>\n",
       "      <td>AL627082.1 - AL512408.1</td>\n",
       "      <td>AL627082.1 - AL512408.1</td>\n",
       "      <td>PIGV - RN7SL165P</td>\n",
       "      <td>PIGV - RN7SL165P</td>\n",
       "      <td>PIGV - RN7SL165P</td>\n",
       "      <td>PIGV - RN7SL165P</td>\n",
       "      <td>KDF1</td>\n",
       "      <td>KDF1</td>\n",
       "      <td>KDF1</td>\n",
       "      <td>...</td>\n",
       "      <td>UBE2L3</td>\n",
       "      <td>UBE2L3 - YDJC</td>\n",
       "      <td>UBE2L3 - YDJC</td>\n",
       "      <td>ZNRF3, ZNRF3-IT1</td>\n",
       "      <td>PLA2G6</td>\n",
       "      <td>PLA2G6</td>\n",
       "      <td>PLA2G6</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>FANCB</td>\n",
       "      <td>FANCB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UPSTREAM_GENE_ID</th>\n",
       "      <td>ENSG00000238290</td>\n",
       "      <td>ENSG00000235069</td>\n",
       "      <td>ENSG00000235069</td>\n",
       "      <td>ENSG00000060642</td>\n",
       "      <td>ENSG00000060642</td>\n",
       "      <td>ENSG00000060642</td>\n",
       "      <td>ENSG00000060642</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000185651</td>\n",
       "      <td>ENSG00000185651</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOWNSTREAM_GENE_ID</th>\n",
       "      <td>ENSG00000229305</td>\n",
       "      <td>ENSG00000260063</td>\n",
       "      <td>ENSG00000260063</td>\n",
       "      <td>ENSG00000241188</td>\n",
       "      <td>ENSG00000241188</td>\n",
       "      <td>ENSG00000241188</td>\n",
       "      <td>ENSG00000241188</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000161179</td>\n",
       "      <td>ENSG00000161179</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP_GENE_IDS</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000175707</td>\n",
       "      <td>ENSG00000175707</td>\n",
       "      <td>ENSG00000175707</td>\n",
       "      <td>...</td>\n",
       "      <td>ENSG00000185651</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000183579, ENSG00000235786</td>\n",
       "      <td>ENSG00000184381</td>\n",
       "      <td>ENSG00000184381</td>\n",
       "      <td>ENSG00000184381</td>\n",
       "      <td>ENSG00000100344</td>\n",
       "      <td>ENSG00000181544</td>\n",
       "      <td>ENSG00000181544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UPSTREAM_GENE_DISTANCE</th>\n",
       "      <td>50508</td>\n",
       "      <td>13474</td>\n",
       "      <td>13474</td>\n",
       "      <td>13504</td>\n",
       "      <td>13504</td>\n",
       "      <td>13504</td>\n",
       "      <td>13504</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3731</td>\n",
       "      <td>3731</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOWNSTREAM_GENE_DISTANCE</th>\n",
       "      <td>16614</td>\n",
       "      <td>30977</td>\n",
       "      <td>30977</td>\n",
       "      <td>2920</td>\n",
       "      <td>2920</td>\n",
       "      <td>2920</td>\n",
       "      <td>2920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>324</td>\n",
       "      <td>324</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STRONGEST SNP-RISK ALLELE</th>\n",
       "      <td>rs149963466-A</td>\n",
       "      <td>rs193084249-A</td>\n",
       "      <td>rs193084249-A</td>\n",
       "      <td>rs12748152-T</td>\n",
       "      <td>rs12748152-T</td>\n",
       "      <td>rs12748152-C</td>\n",
       "      <td>rs12748152-C</td>\n",
       "      <td>rs79598313-C</td>\n",
       "      <td>rs79598313-C</td>\n",
       "      <td>rs79598313-T</td>\n",
       "      <td>...</td>\n",
       "      <td>rs7445-?</td>\n",
       "      <td>rs1811069-T</td>\n",
       "      <td>rs1811069-T</td>\n",
       "      <td>rs8142788-A</td>\n",
       "      <td>rs133027-CT</td>\n",
       "      <td>rs133027-CT</td>\n",
       "      <td>rs2899297-A</td>\n",
       "      <td>rs2294915-T</td>\n",
       "      <td>rs138331350-G</td>\n",
       "      <td>rs138331350-G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNPS</th>\n",
       "      <td>rs149963466</td>\n",
       "      <td>rs193084249</td>\n",
       "      <td>rs193084249</td>\n",
       "      <td>rs12748152</td>\n",
       "      <td>rs12748152</td>\n",
       "      <td>rs12748152</td>\n",
       "      <td>rs12748152</td>\n",
       "      <td>rs79598313</td>\n",
       "      <td>rs79598313</td>\n",
       "      <td>rs79598313</td>\n",
       "      <td>...</td>\n",
       "      <td>rs7445</td>\n",
       "      <td>rs1811069</td>\n",
       "      <td>rs1811069</td>\n",
       "      <td>rs8142788</td>\n",
       "      <td>rs133027</td>\n",
       "      <td>rs133027</td>\n",
       "      <td>rs2899297</td>\n",
       "      <td>rs2294915</td>\n",
       "      <td>rs138331350</td>\n",
       "      <td>rs138331350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MERGED</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNP_ID_CURRENT</th>\n",
       "      <td>1.49963e+08</td>\n",
       "      <td>1.93084e+08</td>\n",
       "      <td>1.93084e+08</td>\n",
       "      <td>1.27482e+07</td>\n",
       "      <td>1.27482e+07</td>\n",
       "      <td>1.27482e+07</td>\n",
       "      <td>1.27482e+07</td>\n",
       "      <td>7.95983e+07</td>\n",
       "      <td>7.95983e+07</td>\n",
       "      <td>7.95983e+07</td>\n",
       "      <td>...</td>\n",
       "      <td>7445</td>\n",
       "      <td>1.81107e+06</td>\n",
       "      <td>1.81107e+06</td>\n",
       "      <td>8.14279e+06</td>\n",
       "      <td>133027</td>\n",
       "      <td>133027</td>\n",
       "      <td>2.8993e+06</td>\n",
       "      <td>2.29492e+06</td>\n",
       "      <td>1.38331e+08</td>\n",
       "      <td>1.38331e+08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CONTEXT</th>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INTERGENIC</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RISK ALLELE FREQUENCY</th>\n",
       "      <td>0.0016</td>\n",
       "      <td>0.98</td>\n",
       "      <td>NR</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.0712</td>\n",
       "      <td>0.925</td>\n",
       "      <td>NR</td>\n",
       "      <td>0.979</td>\n",
       "      <td>NR</td>\n",
       "      <td>0.0216</td>\n",
       "      <td>...</td>\n",
       "      <td>NR</td>\n",
       "      <td>0.801</td>\n",
       "      <td>NR</td>\n",
       "      <td>0.1559</td>\n",
       "      <td>0.56</td>\n",
       "      <td>NR</td>\n",
       "      <td>0.5479</td>\n",
       "      <td>0.246</td>\n",
       "      <td>0.942</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-VALUE</th>\n",
       "      <td>3e-08</td>\n",
       "      <td>2e-14</td>\n",
       "      <td>2e-14</td>\n",
       "      <td>1e-15</td>\n",
       "      <td>9e-16</td>\n",
       "      <td>5e-13</td>\n",
       "      <td>3e-14</td>\n",
       "      <td>4e-14</td>\n",
       "      <td>1e-14</td>\n",
       "      <td>3e-19</td>\n",
       "      <td>...</td>\n",
       "      <td>1e-14</td>\n",
       "      <td>4e-09</td>\n",
       "      <td>1e-13</td>\n",
       "      <td>1e-08</td>\n",
       "      <td>9e-11</td>\n",
       "      <td>2e-12</td>\n",
       "      <td>1e-16</td>\n",
       "      <td>3e-08</td>\n",
       "      <td>3e-11</td>\n",
       "      <td>1e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PVALUE_MLOG</th>\n",
       "      <td>7.52288</td>\n",
       "      <td>13.699</td>\n",
       "      <td>13.699</td>\n",
       "      <td>15</td>\n",
       "      <td>15.0458</td>\n",
       "      <td>12.301</td>\n",
       "      <td>13.5229</td>\n",
       "      <td>13.3979</td>\n",
       "      <td>14</td>\n",
       "      <td>18.5229</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>8.39794</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>10.0458</td>\n",
       "      <td>11.699</td>\n",
       "      <td>16</td>\n",
       "      <td>7.52288</td>\n",
       "      <td>10.5229</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-VALUE (TEXT)</th>\n",
       "      <td>NaN</td>\n",
       "      <td>(EA)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(Trans-ethnic initial)</td>\n",
       "      <td>(EA)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(EA)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(EA)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(EA)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(EA)</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OR or BETA</th>\n",
       "      <td>0.867736</td>\n",
       "      <td>0.113</td>\n",
       "      <td>0.114</td>\n",
       "      <td>0.051</td>\n",
       "      <td>0.0506</td>\n",
       "      <td>0.051</td>\n",
       "      <td>0.052</td>\n",
       "      <td>0.099</td>\n",
       "      <td>0.098</td>\n",
       "      <td>0.0898</td>\n",
       "      <td>...</td>\n",
       "      <td>0.04087</td>\n",
       "      <td>0.028</td>\n",
       "      <td>0.031</td>\n",
       "      <td>0.023</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.025</td>\n",
       "      <td>0.0225</td>\n",
       "      <td>0.0177</td>\n",
       "      <td>0.048</td>\n",
       "      <td>0.049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95% CI (TEXT)</th>\n",
       "      <td>[0.56-1.18] mg/dl increase</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>[NR] unit decrease</td>\n",
       "      <td>[0.038-0.063] unit decrease (EA Beta values)</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>[0.07-0.109] mg dl−1 decrease</td>\n",
       "      <td>...</td>\n",
       "      <td>[0.03-0.051] unit decrease novel</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>[0.015-0.031] mg dl−1 decrease</td>\n",
       "      <td>unit decrease</td>\n",
       "      <td>unit decrease</td>\n",
       "      <td>[0.017-0.028] mg dl−1 decrease</td>\n",
       "      <td>[0.011-0.024] mg dl−1 decrease</td>\n",
       "      <td>unit decrease</td>\n",
       "      <td>unit decrease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PLATFORM [SNPS PASSING QC]</th>\n",
       "      <td>Illumina [24111857] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>NR [NR] (imputed)</td>\n",
       "      <td>Affymetrix, Illumina [~ 1900000] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [up to 31400000] (imputed)</td>\n",
       "      <td>...</td>\n",
       "      <td>Illumina [6108953] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [up to 31400000] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [up to 31400000] (imputed)</td>\n",
       "      <td>Affymetrix [up to 31400000] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNV</th>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAPPED_TRAIT</th>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>...</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAPPED_TRAIT_URI</th>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>...</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <td>GCST004207</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST002223</td>\n",
       "      <td>GCST004232</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST006611</td>\n",
       "      <td>...</td>\n",
       "      <td>GCST006005</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST006611</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST006611</td>\n",
       "      <td>GCST006611</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>GCST007140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENOTYPING TECHNOLOGY</th>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>...</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>38 rows × 1300 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                         0     \\\n",
       "DATE ADDED TO CATALOG                                                 6/21/17   \n",
       "PUBMEDID                                                             28270201   \n",
       "FIRST AUTHOR                                                           Nagy R   \n",
       "DATE                                                                   3/7/17   \n",
       "JOURNAL                                                            Genome Med   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/28270201   \n",
       "STUDY                       Exploration of haplotype research consortium i...   \n",
       "DISEASE/TRAIT                                                 HDL cholesterol   \n",
       "INITIAL SAMPLE SIZE         19,223 British ancestry individuals from 6863 ...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                1p36.23   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                           8,173,210    \n",
       "REPORTED GENE(S)                                                   intergenic   \n",
       "MAPPED_GENE                                           AL034417.2 - AL358876.1   \n",
       "UPSTREAM_GENE_ID                                              ENSG00000238290   \n",
       "DOWNSTREAM_GENE_ID                                            ENSG00000229305   \n",
       "SNP_GENE_IDS                                                              NaN   \n",
       "UPSTREAM_GENE_DISTANCE                                                  50508   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                16614   \n",
       "STRONGEST SNP-RISK ALLELE                                       rs149963466-A   \n",
       "SNPS                                                              rs149963466   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.49963e+08   \n",
       "CONTEXT                                                    intergenic_variant   \n",
       "INTERGENIC                                                                  1   \n",
       "RISK ALLELE FREQUENCY                                                  0.0016   \n",
       "P-VALUE                                                                 3e-08   \n",
       "PVALUE_MLOG                                                           7.52288   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                           0.867736   \n",
       "95% CI (TEXT)                                      [0.56-1.18] mg/dl increase   \n",
       "PLATFORM [SNPS PASSING QC]                      Illumina [24111857] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST004207   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1     \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                1p36.11   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                          26,661,155    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                           AL627082.1 - AL512408.1   \n",
       "UPSTREAM_GENE_ID                                              ENSG00000235069   \n",
       "DOWNSTREAM_GENE_ID                                            ENSG00000260063   \n",
       "SNP_GENE_IDS                                                              NaN   \n",
       "UPSTREAM_GENE_DISTANCE                                                  13474   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                30977   \n",
       "STRONGEST SNP-RISK ALLELE                                       rs193084249-A   \n",
       "SNPS                                                              rs193084249   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.93084e+08   \n",
       "CONTEXT                                                    intergenic_variant   \n",
       "INTERGENIC                                                                  1   \n",
       "RISK ALLELE FREQUENCY                                                    0.98   \n",
       "P-VALUE                                                                 2e-14   \n",
       "PVALUE_MLOG                                                            13.699   \n",
       "P-VALUE (TEXT)                                                           (EA)   \n",
       "OR or BETA                                                              0.113   \n",
       "95% CI (TEXT)                                                   unit increase   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         2     \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                1p36.11   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                          26,661,155    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                           AL627082.1 - AL512408.1   \n",
       "UPSTREAM_GENE_ID                                              ENSG00000235069   \n",
       "DOWNSTREAM_GENE_ID                                            ENSG00000260063   \n",
       "SNP_GENE_IDS                                                              NaN   \n",
       "UPSTREAM_GENE_DISTANCE                                                  13474   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                30977   \n",
       "STRONGEST SNP-RISK ALLELE                                       rs193084249-A   \n",
       "SNPS                                                              rs193084249   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.93084e+08   \n",
       "CONTEXT                                                    intergenic_variant   \n",
       "INTERGENIC                                                                  1   \n",
       "RISK ALLELE FREQUENCY                                                      NR   \n",
       "P-VALUE                                                                 2e-14   \n",
       "PVALUE_MLOG                                                            13.699   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                              0.114   \n",
       "95% CI (TEXT)                                                   unit increase   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         3     \\\n",
       "DATE ADDED TO CATALOG                                                 5/12/14   \n",
       "PUBMEDID                                                             24097068   \n",
       "FIRST AUTHOR                                                        Willer CJ   \n",
       "DATE                                                                  10/6/13   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/24097068   \n",
       "STUDY                       Discovery and refinement of loci associated wi...   \n",
       "DISEASE/TRAIT                                                 HDL cholesterol   \n",
       "INITIAL SAMPLE SIZE                      94,595 European ancestry individuals   \n",
       "REPLICATION SAMPLE SIZE                  93,982 European ancestry individuals   \n",
       "REGION                                                                1p36.11   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                          26,811,902    \n",
       "REPORTED GENE(S)                                                  PIGV, NR0B2   \n",
       "MAPPED_GENE                                                  PIGV - RN7SL165P   \n",
       "UPSTREAM_GENE_ID                                              ENSG00000060642   \n",
       "DOWNSTREAM_GENE_ID                                            ENSG00000241188   \n",
       "SNP_GENE_IDS                                                              NaN   \n",
       "UPSTREAM_GENE_DISTANCE                                                  13504   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                 2920   \n",
       "STRONGEST SNP-RISK ALLELE                                        rs12748152-T   \n",
       "SNPS                                                               rs12748152   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.27482e+07   \n",
       "CONTEXT                                                    intergenic_variant   \n",
       "INTERGENIC                                                                  1   \n",
       "RISK ALLELE FREQUENCY                                                    0.09   \n",
       "P-VALUE                                                                 1e-15   \n",
       "PVALUE_MLOG                                                                15   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                              0.051   \n",
       "95% CI (TEXT)                                              [NR] unit decrease   \n",
       "PLATFORM [SNPS PASSING QC]                                  NR [NR] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST002223   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         4     \\\n",
       "DATE ADDED TO CATALOG                                                 6/26/17   \n",
       "PUBMEDID                                                             28334899   \n",
       "FIRST AUTHOR                                                     Spracklen CN   \n",
       "DATE                                                                  2/21/17   \n",
       "JOURNAL                                                         Hum Mol Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/28334899   \n",
       "STUDY                       Association analyses of East Asian individuals...   \n",
       "DISEASE/TRAIT                                          HDL cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         34,930 East Asian ancestry individuals, 187,16...   \n",
       "REPLICATION SAMPLE SIZE                    8,741 Chinese ancestry individuals   \n",
       "REGION                                                                1p36.11   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                          26,811,902    \n",
       "REPORTED GENE(S)                                                  PIGV, NROB2   \n",
       "MAPPED_GENE                                                  PIGV - RN7SL165P   \n",
       "UPSTREAM_GENE_ID                                              ENSG00000060642   \n",
       "DOWNSTREAM_GENE_ID                                            ENSG00000241188   \n",
       "SNP_GENE_IDS                                                              NaN   \n",
       "UPSTREAM_GENE_DISTANCE                                                  13504   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                 2920   \n",
       "STRONGEST SNP-RISK ALLELE                                        rs12748152-T   \n",
       "SNPS                                                               rs12748152   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.27482e+07   \n",
       "CONTEXT                                                    intergenic_variant   \n",
       "INTERGENIC                                                                  1   \n",
       "RISK ALLELE FREQUENCY                                                  0.0712   \n",
       "P-VALUE                                                                 9e-16   \n",
       "PVALUE_MLOG                                                           15.0458   \n",
       "P-VALUE (TEXT)                                         (Trans-ethnic initial)   \n",
       "OR or BETA                                                             0.0506   \n",
       "95% CI (TEXT)                    [0.038-0.063] unit decrease (EA Beta values)   \n",
       "PLATFORM [SNPS PASSING QC]         Affymetrix, Illumina [~ 1900000] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST004232   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         5     \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                1p36.11   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                          26,811,902    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                                  PIGV - RN7SL165P   \n",
       "UPSTREAM_GENE_ID                                              ENSG00000060642   \n",
       "DOWNSTREAM_GENE_ID                                            ENSG00000241188   \n",
       "SNP_GENE_IDS                                                              NaN   \n",
       "UPSTREAM_GENE_DISTANCE                                                  13504   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                 2920   \n",
       "STRONGEST SNP-RISK ALLELE                                        rs12748152-C   \n",
       "SNPS                                                               rs12748152   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.27482e+07   \n",
       "CONTEXT                                                    intergenic_variant   \n",
       "INTERGENIC                                                                  1   \n",
       "RISK ALLELE FREQUENCY                                                   0.925   \n",
       "P-VALUE                                                                 5e-13   \n",
       "PVALUE_MLOG                                                            12.301   \n",
       "P-VALUE (TEXT)                                                           (EA)   \n",
       "OR or BETA                                                              0.051   \n",
       "95% CI (TEXT)                                                   unit increase   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         6     \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                1p36.11   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                          26,811,902    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                                  PIGV - RN7SL165P   \n",
       "UPSTREAM_GENE_ID                                              ENSG00000060642   \n",
       "DOWNSTREAM_GENE_ID                                            ENSG00000241188   \n",
       "SNP_GENE_IDS                                                              NaN   \n",
       "UPSTREAM_GENE_DISTANCE                                                  13504   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                 2920   \n",
       "STRONGEST SNP-RISK ALLELE                                        rs12748152-C   \n",
       "SNPS                                                               rs12748152   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.27482e+07   \n",
       "CONTEXT                                                    intergenic_variant   \n",
       "INTERGENIC                                                                  1   \n",
       "RISK ALLELE FREQUENCY                                                      NR   \n",
       "P-VALUE                                                                 3e-14   \n",
       "PVALUE_MLOG                                                           13.5229   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                              0.052   \n",
       "95% CI (TEXT)                                                   unit increase   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         7     \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                1p36.11   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                          26,958,422    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                                              KDF1   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                                  ENSG00000175707   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                        rs79598313-C   \n",
       "SNPS                                                               rs79598313   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    7.95983e+07   \n",
       "CONTEXT                                                        intron_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                   0.979   \n",
       "P-VALUE                                                                 4e-14   \n",
       "PVALUE_MLOG                                                           13.3979   \n",
       "P-VALUE (TEXT)                                                           (EA)   \n",
       "OR or BETA                                                              0.099   \n",
       "95% CI (TEXT)                                                   unit increase   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         8     \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                1p36.11   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                          26,958,422    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                                              KDF1   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                                  ENSG00000175707   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                        rs79598313-C   \n",
       "SNPS                                                               rs79598313   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    7.95983e+07   \n",
       "CONTEXT                                                        intron_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                      NR   \n",
       "P-VALUE                                                                 1e-14   \n",
       "PVALUE_MLOG                                                                14   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                              0.098   \n",
       "95% CI (TEXT)                                                   unit increase   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         9     \\\n",
       "DATE ADDED TO CATALOG                                                11/26/18   \n",
       "PUBMEDID                                                             30275531   \n",
       "FIRST AUTHOR                                                         Klarin D   \n",
       "DATE                                                                  10/1/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/30275531   \n",
       "STUDY                       Genetics of blood lipids among ~300,000 multi-...   \n",
       "DISEASE/TRAIT                                                 HDL cholesterol   \n",
       "INITIAL SAMPLE SIZE         215,551 European ancestry individuals, 57,332 ...   \n",
       "REPLICATION SAMPLE SIZE                             Up to 319,677 individuals   \n",
       "REGION                                                                1p36.11   \n",
       "CHR_ID                                                                      1   \n",
       " CHR_POS                                                          26,958,422    \n",
       "REPORTED GENE(S)                                                     C1orf172   \n",
       "MAPPED_GENE                                                              KDF1   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                                  ENSG00000175707   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                        rs79598313-T   \n",
       "SNPS                                                               rs79598313   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    7.95983e+07   \n",
       "CONTEXT                                                        intron_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                  0.0216   \n",
       "P-VALUE                                                                 3e-19   \n",
       "PVALUE_MLOG                                                           18.5229   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                             0.0898   \n",
       "95% CI (TEXT)                                   [0.07-0.109] mg dl−1 decrease   \n",
       "PLATFORM [SNPS PASSING QC]              Affymetrix [up to 31400000] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST006611   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                            ...  \\\n",
       "DATE ADDED TO CATALOG       ...   \n",
       "PUBMEDID                    ...   \n",
       "FIRST AUTHOR                ...   \n",
       "DATE                        ...   \n",
       "JOURNAL                     ...   \n",
       "LINK                        ...   \n",
       "STUDY                       ...   \n",
       "DISEASE/TRAIT               ...   \n",
       "INITIAL SAMPLE SIZE         ...   \n",
       "REPLICATION SAMPLE SIZE     ...   \n",
       "REGION                      ...   \n",
       "CHR_ID                      ...   \n",
       " CHR_POS                    ...   \n",
       "REPORTED GENE(S)            ...   \n",
       "MAPPED_GENE                 ...   \n",
       "UPSTREAM_GENE_ID            ...   \n",
       "DOWNSTREAM_GENE_ID          ...   \n",
       "SNP_GENE_IDS                ...   \n",
       "UPSTREAM_GENE_DISTANCE      ...   \n",
       "DOWNSTREAM_GENE_DISTANCE    ...   \n",
       "STRONGEST SNP-RISK ALLELE   ...   \n",
       "SNPS                        ...   \n",
       "MERGED                      ...   \n",
       "SNP_ID_CURRENT              ...   \n",
       "CONTEXT                     ...   \n",
       "INTERGENIC                  ...   \n",
       "RISK ALLELE FREQUENCY       ...   \n",
       "P-VALUE                     ...   \n",
       "PVALUE_MLOG                 ...   \n",
       "P-VALUE (TEXT)              ...   \n",
       "OR or BETA                  ...   \n",
       "95% CI (TEXT)               ...   \n",
       "PLATFORM [SNPS PASSING QC]  ...   \n",
       "CNV                         ...   \n",
       "MAPPED_TRAIT                ...   \n",
       "MAPPED_TRAIT_URI            ...   \n",
       "STUDY ACCESSION             ...   \n",
       "GENOTYPING TECHNOLOGY       ...   \n",
       "\n",
       "                                                                         1290  \\\n",
       "DATE ADDED TO CATALOG                                                 7/28/18   \n",
       "PUBMEDID                                                             29403010   \n",
       "FIRST AUTHOR                                                          Kanai M   \n",
       "DATE                                                                   2/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29403010   \n",
       "STUDY                       Genetic analysis of quantitative traits in the...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE                      70,657 Japanese ancestry individuals   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                               22q11.21   \n",
       "CHR_ID                                                                     22   \n",
       " CHR_POS                                                          21,622,758    \n",
       "REPORTED GENE(S)                                                       UBE2L3   \n",
       "MAPPED_GENE                                                            UBE2L3   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                                  ENSG00000185651   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                            rs7445-?   \n",
       "SNPS                                                                   rs7445   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                           7445   \n",
       "CONTEXT                                                   3_prime_UTR_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                      NR   \n",
       "P-VALUE                                                                 1e-14   \n",
       "PVALUE_MLOG                                                                14   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                            0.04087   \n",
       "95% CI (TEXT)                                [0.03-0.051] unit decrease novel   \n",
       "PLATFORM [SNPS PASSING QC]                       Illumina [6108953] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST006005   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1291  \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                               22q11.21   \n",
       "CHR_ID                                                                     22   \n",
       " CHR_POS                                                          21,627,765    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                                     UBE2L3 - YDJC   \n",
       "UPSTREAM_GENE_ID                                              ENSG00000185651   \n",
       "DOWNSTREAM_GENE_ID                                            ENSG00000161179   \n",
       "SNP_GENE_IDS                                                              NaN   \n",
       "UPSTREAM_GENE_DISTANCE                                                   3731   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  324   \n",
       "STRONGEST SNP-RISK ALLELE                                         rs1811069-T   \n",
       "SNPS                                                                rs1811069   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.81107e+06   \n",
       "CONTEXT                                                    intergenic_variant   \n",
       "INTERGENIC                                                                  1   \n",
       "RISK ALLELE FREQUENCY                                                   0.801   \n",
       "P-VALUE                                                                 4e-09   \n",
       "PVALUE_MLOG                                                           8.39794   \n",
       "P-VALUE (TEXT)                                                           (EA)   \n",
       "OR or BETA                                                              0.028   \n",
       "95% CI (TEXT)                                                   unit increase   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1292  \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                               22q11.21   \n",
       "CHR_ID                                                                     22   \n",
       " CHR_POS                                                          21,627,765    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                                     UBE2L3 - YDJC   \n",
       "UPSTREAM_GENE_ID                                              ENSG00000185651   \n",
       "DOWNSTREAM_GENE_ID                                            ENSG00000161179   \n",
       "SNP_GENE_IDS                                                              NaN   \n",
       "UPSTREAM_GENE_DISTANCE                                                   3731   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  324   \n",
       "STRONGEST SNP-RISK ALLELE                                         rs1811069-T   \n",
       "SNPS                                                                rs1811069   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.81107e+06   \n",
       "CONTEXT                                                    intergenic_variant   \n",
       "INTERGENIC                                                                  1   \n",
       "RISK ALLELE FREQUENCY                                                      NR   \n",
       "P-VALUE                                                                 1e-13   \n",
       "PVALUE_MLOG                                                                13   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                              0.031   \n",
       "95% CI (TEXT)                                                   unit increase   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1293  \\\n",
       "DATE ADDED TO CATALOG                                                11/26/18   \n",
       "PUBMEDID                                                             30275531   \n",
       "FIRST AUTHOR                                                         Klarin D   \n",
       "DATE                                                                  10/1/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/30275531   \n",
       "STUDY                       Genetics of blood lipids among ~300,000 multi-...   \n",
       "DISEASE/TRAIT                                                 HDL cholesterol   \n",
       "INITIAL SAMPLE SIZE         215,551 European ancestry individuals, 57,332 ...   \n",
       "REPLICATION SAMPLE SIZE                             Up to 319,677 individuals   \n",
       "REGION                                                                22q12.1   \n",
       "CHR_ID                                                                     22   \n",
       " CHR_POS                                                          29,004,527    \n",
       "REPORTED GENE(S)                                                        ZNRF3   \n",
       "MAPPED_GENE                                                  ZNRF3, ZNRF3-IT1   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                 ENSG00000183579, ENSG00000235786   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                         rs8142788-A   \n",
       "SNPS                                                                rs8142788   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    8.14279e+06   \n",
       "CONTEXT                                                        intron_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                  0.1559   \n",
       "P-VALUE                                                                 1e-08   \n",
       "PVALUE_MLOG                                                                 8   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                              0.023   \n",
       "95% CI (TEXT)                                  [0.015-0.031] mg dl−1 decrease   \n",
       "PLATFORM [SNPS PASSING QC]              Affymetrix [up to 31400000] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST006611   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1294  \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                22q13.1   \n",
       "CHR_ID                                                                     22   \n",
       " CHR_POS                                                          38,179,492    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                                            PLA2G6   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                                  ENSG00000184381   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                         rs133027-CT   \n",
       "SNPS                                                                 rs133027   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                         133027   \n",
       "CONTEXT                                                        intron_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                    0.56   \n",
       "P-VALUE                                                                 9e-11   \n",
       "PVALUE_MLOG                                                           10.0458   \n",
       "P-VALUE (TEXT)                                                           (EA)   \n",
       "OR or BETA                                                              0.025   \n",
       "95% CI (TEXT)                                                   unit decrease   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1295  \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                22q13.1   \n",
       "CHR_ID                                                                     22   \n",
       " CHR_POS                                                          38,179,492    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                                            PLA2G6   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                                  ENSG00000184381   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                         rs133027-CT   \n",
       "SNPS                                                                 rs133027   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                         133027   \n",
       "CONTEXT                                                        intron_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                      NR   \n",
       "P-VALUE                                                                 2e-12   \n",
       "PVALUE_MLOG                                                            11.699   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                              0.025   \n",
       "95% CI (TEXT)                                                   unit decrease   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1296  \\\n",
       "DATE ADDED TO CATALOG                                                11/26/18   \n",
       "PUBMEDID                                                             30275531   \n",
       "FIRST AUTHOR                                                         Klarin D   \n",
       "DATE                                                                  10/1/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/30275531   \n",
       "STUDY                       Genetics of blood lipids among ~300,000 multi-...   \n",
       "DISEASE/TRAIT                                                 HDL cholesterol   \n",
       "INITIAL SAMPLE SIZE         215,551 European ancestry individuals, 57,332 ...   \n",
       "REPLICATION SAMPLE SIZE                             Up to 319,677 individuals   \n",
       "REGION                                                                22q13.1   \n",
       "CHR_ID                                                                     22   \n",
       " CHR_POS                                                          38,198,661    \n",
       "REPORTED GENE(S)                                                         MAFF   \n",
       "MAPPED_GENE                                                            PLA2G6   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                                  ENSG00000184381   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                         rs2899297-A   \n",
       "SNPS                                                                rs2899297   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                     2.8993e+06   \n",
       "CONTEXT                                                        intron_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                  0.5479   \n",
       "P-VALUE                                                                 1e-16   \n",
       "PVALUE_MLOG                                                                16   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                             0.0225   \n",
       "95% CI (TEXT)                                  [0.017-0.028] mg dl−1 decrease   \n",
       "PLATFORM [SNPS PASSING QC]              Affymetrix [up to 31400000] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST006611   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1297  \\\n",
       "DATE ADDED TO CATALOG                                                11/26/18   \n",
       "PUBMEDID                                                             30275531   \n",
       "FIRST AUTHOR                                                         Klarin D   \n",
       "DATE                                                                  10/1/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/30275531   \n",
       "STUDY                       Genetics of blood lipids among ~300,000 multi-...   \n",
       "DISEASE/TRAIT                                                 HDL cholesterol   \n",
       "INITIAL SAMPLE SIZE         215,551 European ancestry individuals, 57,332 ...   \n",
       "REPLICATION SAMPLE SIZE                             Up to 319,677 individuals   \n",
       "REGION                                                               22q13.31   \n",
       "CHR_ID                                                                     22   \n",
       " CHR_POS                                                          43,945,024    \n",
       "REPORTED GENE(S)                                                       PNPLA3   \n",
       "MAPPED_GENE                                                            PNPLA3   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                                  ENSG00000100344   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                         rs2294915-T   \n",
       "SNPS                                                                rs2294915   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    2.29492e+06   \n",
       "CONTEXT                                                        intron_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                   0.246   \n",
       "P-VALUE                                                                 3e-08   \n",
       "PVALUE_MLOG                                                           7.52288   \n",
       "P-VALUE (TEXT)                                                            NaN   \n",
       "OR or BETA                                                             0.0177   \n",
       "95% CI (TEXT)                                  [0.011-0.024] mg dl−1 decrease   \n",
       "PLATFORM [SNPS PASSING QC]              Affymetrix [up to 31400000] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST006611   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1298  \\\n",
       "DATE ADDED TO CATALOG                                                 2/14/19   \n",
       "PUBMEDID                                                             29507422   \n",
       "FIRST AUTHOR                                                      Hoffmann TJ   \n",
       "DATE                                                                   3/5/18   \n",
       "JOURNAL                                                             Nat Genet   \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "STUDY                       A large electronic-health-record-based genome-...   \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels   \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...   \n",
       "REPLICATION SAMPLE SIZE                                                   NaN   \n",
       "REGION                                                                 Xp22.2   \n",
       "CHR_ID                                                                      X   \n",
       " CHR_POS                                                          14,841,981    \n",
       "REPORTED GENE(S)                                                           NR   \n",
       "MAPPED_GENE                                                             FANCB   \n",
       "UPSTREAM_GENE_ID                                                          NaN   \n",
       "DOWNSTREAM_GENE_ID                                                        NaN   \n",
       "SNP_GENE_IDS                                                  ENSG00000181544   \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN   \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN   \n",
       "STRONGEST SNP-RISK ALLELE                                       rs138331350-G   \n",
       "SNPS                                                              rs138331350   \n",
       "MERGED                                                                      0   \n",
       "SNP_ID_CURRENT                                                    1.38331e+08   \n",
       "CONTEXT                                                        intron_variant   \n",
       "INTERGENIC                                                                  0   \n",
       "RISK ALLELE FREQUENCY                                                   0.942   \n",
       "P-VALUE                                                                 3e-11   \n",
       "PVALUE_MLOG                                                           10.5229   \n",
       "P-VALUE (TEXT)                                                           (EA)   \n",
       "OR or BETA                                                              0.048   \n",
       "95% CI (TEXT)                                                   unit decrease   \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)   \n",
       "CNV                                                                         N   \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement   \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612   \n",
       "STUDY ACCESSION                                                    GCST007140   \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array   \n",
       "\n",
       "                                                                         1299  \n",
       "DATE ADDED TO CATALOG                                                 2/14/19  \n",
       "PUBMEDID                                                             29507422  \n",
       "FIRST AUTHOR                                                      Hoffmann TJ  \n",
       "DATE                                                                   3/5/18  \n",
       "JOURNAL                                                             Nat Genet  \n",
       "LINK                                     www.ncbi.nlm.nih.gov/pubmed/29507422  \n",
       "STUDY                       A large electronic-health-record-based genome-...  \n",
       "DISEASE/TRAIT                     High density lipoprotein cholesterol levels  \n",
       "INITIAL SAMPLE SIZE         76,627 European ancestry individuals, 7,795 Hi...  \n",
       "REPLICATION SAMPLE SIZE                                                   NaN  \n",
       "REGION                                                                 Xp22.2  \n",
       "CHR_ID                                                                      X  \n",
       " CHR_POS                                                          14,841,981   \n",
       "REPORTED GENE(S)                                                           NR  \n",
       "MAPPED_GENE                                                             FANCB  \n",
       "UPSTREAM_GENE_ID                                                          NaN  \n",
       "DOWNSTREAM_GENE_ID                                                        NaN  \n",
       "SNP_GENE_IDS                                                  ENSG00000181544  \n",
       "UPSTREAM_GENE_DISTANCE                                                    NaN  \n",
       "DOWNSTREAM_GENE_DISTANCE                                                  NaN  \n",
       "STRONGEST SNP-RISK ALLELE                                       rs138331350-G  \n",
       "SNPS                                                              rs138331350  \n",
       "MERGED                                                                      0  \n",
       "SNP_ID_CURRENT                                                    1.38331e+08  \n",
       "CONTEXT                                                        intron_variant  \n",
       "INTERGENIC                                                                  0  \n",
       "RISK ALLELE FREQUENCY                                                      NR  \n",
       "P-VALUE                                                                 1e-12  \n",
       "PVALUE_MLOG                                                                12  \n",
       "P-VALUE (TEXT)                                                            NaN  \n",
       "OR or BETA                                                              0.049  \n",
       "95% CI (TEXT)                                                   unit decrease  \n",
       "PLATFORM [SNPS PASSING QC]            Affymetrix [at least 7091467] (imputed)  \n",
       "CNV                                                                         N  \n",
       "MAPPED_TRAIT                 high density lipoprotein cholesterol measurement  \n",
       "MAPPED_TRAIT_URI                         http://www.ebi.ac.uk/efo/EFO_0004612  \n",
       "STUDY ACCESSION                                                    GCST007140  \n",
       "GENOTYPING TECHNOLOGY                            Genome-wide genotyping array  \n",
       "\n",
       "[38 rows x 1300 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "novel_df.T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
