{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Notes: changed calculation of expected p values according to Anna's code\n",
    "from scipy.stats import chi2\n",
    "from scipy.stats import beta\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "import urllib\n",
    "\n",
    "# Read a .txt or .csv file containing SNPs and P values to be plotted\n",
    "# Function checks delimiters automatically\n",
    "# Parameter: filename - .txt or .csv file containing information needed\n",
    "# Return: pandas dataframe of the file\n",
    "def __read_file(filename):\n",
    "    fh = open(filename, 'r')\n",
    "    line = fh.readline()\n",
    "    if line=='': return None # Empty file, return None\n",
    "    \n",
    "    tmp = line.strip().split(',')\n",
    "\n",
    "    # Create an empty dataframe, read data into df later\n",
    "    df = pd.DataFrame()\n",
    "    # Check delimiter type, \",\" or \" \"\n",
    "    # One of the read_csv() call should work\n",
    "    if len(tmp) == 1:\n",
    "        try: df = pd.read_csv(filename, sep='\\t')\n",
    "        except pd.errors.ParserError: pass\n",
    "#              print('Try 2 catched')     \n",
    "        \n",
    "        try: df = pd.read_csv(filename, delim_whitespace=True)\n",
    "        except pd.errors.ParserError: pass\n",
    "#             print('Try 2 catched')\n",
    "#         File format maybe wrong if both read_csv failed\n",
    "\n",
    "    else:\n",
    "        df = pd.read_csv(filename)\n",
    "    \n",
    "    # File only has a line of tile, return None  \n",
    "    if len(df)==0: return None \n",
    "    \n",
    "    return(df)\n",
    "\n",
    "# ====================================================================\n",
    "# Return -log10 transformed observed and expected p vlaues of a dataset(dataframe)\n",
    "# Saved some old code for future reference\n",
    "def __log_obsv_and_expt(df, column_title):\n",
    "    scale = 0.5/np.log(10)\n",
    "    Pobsv = df.loc[:, column_title]  # Observed p values\n",
    "    Pobsv = np.sort(Pobsv.dropna().values)\n",
    "    \n",
    "    # Calculate expected p\n",
    "    # A method from qqman package. Refer to ppooints function in R.\n",
    "    # The results are slightly differetn from Anna's\n",
    "#     if len(Pobsv)>10:\n",
    "#         Pexpt = (np.arange(1, len(Pobsv)+1)-0.5) / (len(Pobsv)+1-1)\n",
    "#     else:\n",
    "#         Pexpt = (np.arange(1, len(Pobsv)+1)-3.0/8) / (len(Pobsv)+1-2*3.0/8)\n",
    "        \n",
    "        # From Anna's code, she used this one to calculate chi-squared test statistics, but not for logP qq plot\n",
    "#         Pexpt = chi2.ppf(np.arange(1, len(Pobsv) + 1, 1) / (len(Pobsv) + 1), df=1)\n",
    "\n",
    "    ln_Pexpt = 2*np.cumsum(1/np.arange(len(Pobsv), 0, -1)) # -2*ln(Pexpt), from Anna's code, don't understand why\n",
    "    logPobsv = -np.log10(Pobsv)\n",
    "    logPexpt = ln_Pexpt * scale # Convert from -2ln(x) to -log(x). from Anna's code\n",
    "    \n",
    "    return(Pobsv, np.sort(logPobsv), logPexpt)\n",
    "\n",
    "\n",
    "# Calculate and return lambda (inflation)\n",
    "# This is different from Anna's code, which used mean(obsvd)/mean(expctd)?\n",
    "def __get_inflation__(observed):\n",
    "    obsv_median = np.median(observed)\n",
    "    Chi = chi2.ppf(1.0 - obsv_median, 1) # 1 refers to freedom of 1\n",
    "    lmbd = Chi / chi2.ppf(0.5, 1)\n",
    "    return lmbd\n",
    "\n",
    "\n",
    "# Plot confidence interval\n",
    "def __plot_ci(ax, ci, logPexpt):\n",
    "    n = len(logPexpt)\n",
    "    df = 2 # Degree of freedom, from Anna's code (Why 2?)\n",
    "    # From Anna's code. Scale is used to convert 2ln(x) to log10(x), but I don't understand it\n",
    "    scale = 0.5/np.log(10)\n",
    "\n",
    "    a = list(range(1, n+1))\n",
    "    b = list(range(n, 0, -1))\n",
    "\n",
    "    clower = beta.ppf(q=(1-ci)/2, a = a, b = b)\n",
    "    cupper = beta.ppf(q=1-(1-ci)/2, a = a, b = b)\n",
    "    lower_bond = chi2.ppf(q=clower, df=df)\n",
    "    upper_bond = chi2.ppf(q=cupper, df=df)\n",
    "\n",
    "    plt.fill_between(x=logPexpt, y1=lower_bond*scale, y2=upper_bond*scale,\n",
    "                     alpha=0.2, color='k', linewidth=0, zorder=0) # Plot at the bottom\n",
    "\n",
    "    \n",
    "# The tricky part for me is to query haploreg database thorugh their website.\n",
    "# Inspecct the element of the website to find out names of each variable\n",
    "# Recommand using Chrome, go to developer mode, refresh the website under Network tab\n",
    "# This function is intended to query less than 1000 SNPs at a time, otherwise HaploReg may not be happy\n",
    "# Return: a set of rsIDs of accociated SNPs (and the queried SNP)\n",
    "\n",
    "# Parameters needed to send to haploreg website, and some explaination\n",
    "# input_snps (query): input SNPs in a string, each SNP is separated by ','\n",
    "#        SNP number per query is set to be 1000, otherwise it takes too long and haploreg may refuse to do it.\n",
    "#        15000 SNPs did not work for test run.\n",
    "# gwas_id: the dropdown list to choose GWAS study, when no file or query SNP(s) is provided\n",
    "# r2_threshold (ldThresh): r^2 threshold, default is 0.2 in this code\n",
    "# ldPop: 1000G Phase 1 population for LD calculation. Other options includes AFR, AMR and ASN.\n",
    "# epi: Source for epigenomes\n",
    "# cons: Mammalian conservation algorithm. 'siphy'=SiPhy-omega, 'gerp'=GERP, or 'both'\n",
    "# genetypes: Show position relative to\n",
    "# output: set output result type to 'text' for python code to process\n",
    "def query_haploreg(input_snp,\n",
    "                   r2_threshold=0.2,\n",
    "                   ldPop='EUR',\n",
    "                   epi='vanilla',\n",
    "                   cons='siphy',\n",
    "                   genetypes='gencode'):\n",
    "\n",
    "    params_library = {'query':input_snp,\n",
    "                      'gwas_id':0,\n",
    "                      'ldThresh': r2_threshold,\n",
    "                      'ldPop': ldPop,\n",
    "                      'epi': epi,\n",
    "                      'cons': cons,\n",
    "                      'genetypes': genetypes,\n",
    "                      'output':'text'}\n",
    "    \n",
    "    # parameters passed to the website, needs to be parsed and in binary\n",
    "    params = urllib.parse.urlencode(params_library).encode(\"utf-8\")\n",
    "    # url of HaploReg4.1\n",
    "    url = 'https://pubs.broadinstitute.org/mammals/haploreg/haploreg.php'\n",
    "    # Query with parameters\n",
    "    query = urllib.request.urlopen(url, params)\n",
    "    \n",
    "    # Keep console running\n",
    "    print('-- Waiting for HaploReg to return results...')\n",
    "    \n",
    "    content = query.read().decode(\"utf-8\")\n",
    "    # Find accociated SNPs in the content returned from HaploReg\n",
    "    matches = re.findall('rs[0-9]+', content)\n",
    "\n",
    "    # Return unique ones of the query\n",
    "    return(list(set(matches)))\n",
    "\n",
    "\n",
    "# This function returns known SNPs (and associated SNPs) based on user input file\n",
    "# Ideally know SNPs should not be more than 1500, otherwise it is too much on HaploReg v4.1\n",
    "# Parameters:\n",
    "#  - filename: a file contains known SNPs, provided by user\n",
    "#  - r2_threshold: LD threshold, defualt is 0.2\n",
    "#  - SNPs_column_title: Column title of SNPs in that file\n",
    "# Return: a set of associated SNPs pulled from HaploReg and known SNPs provided by user\n",
    "def __known_SNPs(filename, ldPop, r2_threshold=0.2, known_SNPs_column_title='SNP'):\n",
    "    # Keep console running\n",
    "    print('Getting associated SNPs of known SNPs from HaploReg')\n",
    "    \n",
    "    known_SNPs_df = __read_file(filename) # Dataframe of known SNPs\n",
    "    \n",
    "    # Split into smaller lists to query HaploReg (100 SNPs each query)\n",
    "    count = 0\n",
    "    split_point = 99\n",
    "    associated_and_known_SNPs = [] # Store queried results from HaploReg in a list\n",
    "    if len(known_SNPs_df)>100:\n",
    "        for i in range(0,len(known_SNPs_df), 100):\n",
    "            # Get SNPs to be queried, join them by ',' and query together\n",
    "            known_SNPs = ','.join(known_SNPs_df.loc[:, known_SNPs_column_title].values[i:split_point])\n",
    "            associated_and_known_SNPs = associated_and_known_SNPs + query_haploreg(input_snp=known_SNPs,\n",
    "                                                                                   r2_threshold=r2_threshold,\n",
    "                                                                                   ldPop=ldPop)\n",
    "            \n",
    "            print('  -- Number of SNPs queried: ' +  str((split_point+1)))\n",
    "            \n",
    "            if split_point+100 < len(known_SNPs_df):\n",
    "                split_point = split_point+100\n",
    "            else: split_point = len(known_SNPs_df)\n",
    "\n",
    "    # Remove duplicates of queried results and return\n",
    "    return(set(associated_and_known_SNPs))\n",
    "\n",
    "\n",
    "# This function removes known SNPs and save novel SNPs in a .csv file\n",
    "# Parameters:\n",
    "#  - original_df: a dataframe containing raw data of SNPs and p values\n",
    "#  - known_SNPs: a list of SNPs found in other GWAS, that are associated with traits of interest\n",
    "#  - known_SNPs_column_title: Column title of SNPs in that file\n",
    "# Return: a datafram of novel SNPs and p values\n",
    "def __remove_known_SNPs(original_df, known_SNPs,all_SNPs_column_title='SNP',\n",
    "                        novel_filename='novel_SNPs.csv'):\n",
    "    # Keep console running\n",
    "    print('Removing known SNPs...')\n",
    "    \n",
    "    # Join the original dataframe and konw SNPs, output novel SNPs to be plotted\n",
    "    original_df.set_index(all_SNPs_column_title, inplace=True)\n",
    "    known_SNPs_df = pd.DataFrame([], index=known_SNPs)\n",
    "    to_be_removed_SNPs_df = original_df.join(known_SNPs_df, how='inner')\n",
    "    novel_df = original_df.drop(to_be_removed_SNPs_df.index.values)\n",
    "    \n",
    "    novel_df.to_csv(novel_filename)\n",
    "    return(novel_df)\n",
    "\n",
    "\n",
    "# Parameters:\n",
    "# - filename: name of input file that contains all SNPs discovered, should starts with column titles not empty line\n",
    "# - output: name of the output file to save figure of the QQ plot\n",
    "# - p_value_column_title: title of the column that contains P values in the input file\n",
    "# - title, xlabel, ylabel, dpi=300: label and resolution parameters of the figure\n",
    "# - plot_novel: a plot of novel SNPs will be plotted with original SNPs if this is set to true.\n",
    "#               User needs to supply a known SNPs file in order to look for accociated SNPs.\n",
    "# - known_SNPs_filename:\n",
    "# - novel_filename: \n",
    "# - SNPs_column_title: \n",
    "# - r2_threshold: LD threshold to query HaploReg for associated SNPs\n",
    "# - ldPop: population for LD calculation, defult is European. Other options includes AFR, AMR and ASN.\n",
    "# - plot_ci: plot shaded area of confidence interval if true\n",
    "# - ci: confidence interval, default is 0.95\n",
    "#\n",
    "# Returns: (fig, ax, lambda)\n",
    "# - fig and ax for more custermizations\n",
    "def qqplot(filename,\n",
    "           output='output.png',\n",
    "           p_value_column_title = 'P',\n",
    "           title='Q-Q plot',\n",
    "           xlabel='Expected –log10 P-values',\n",
    "           ylabel='Observed –log10 P-values',\n",
    "           dpi=300,\n",
    "           plot_novel = False,\n",
    "           known_SNPs_filename='',\n",
    "           novel_filename = 'novel_SNPs.csv',\n",
    "           known_SNPs_column_title = 'SNP',\n",
    "           all_SNPs_column_title = 'SNP',\n",
    "           r2_threshold=0.2,\n",
    "           ldPop='EUR',\n",
    "           plot_ci=True,\n",
    "           ci=0.95):\n",
    "    # Read in file, calsulate -log10 p values and lambda\n",
    "    df = __read_file(filename)\n",
    "    \n",
    "    if df is None:\n",
    "        print('Empty file, nothing is plotted')\n",
    "        return(None, None, None) # If file is empty or only has titles, return directly\n",
    "    \n",
    "    fig, ax = plt.subplots(dpi=dpi)\n",
    "    \n",
    "    # Plot all SNPs as dots\n",
    "    Pobsv, logPobsv, logPexpt = __log_obsv_and_expt(df, p_value_column_title)\n",
    "    infl_original = __get_inflation__(Pobsv)\n",
    "    \n",
    "    ax.plot(logPexpt, logPexpt, color='r', linewidth=0.4)\n",
    "    ax.plot(logPexpt, logPobsv, linestyle='', marker='o', markersize=2, markeredgewidth=0.5,\n",
    "            fillstyle='none', color='k', zorder=3)\n",
    "    \n",
    "    # Plot shaded area of confidence intervel\n",
    "    if plot_ci==True:\n",
    "        if ci >= 1: print(\"Confidence intervel >= 1, not ci plotted\")\n",
    "        elif ci <= 0: print(\"Confidence intervel <= 0, not ci plotted\")\n",
    "        else: __plot_ci(ax, ci, logPexpt)\n",
    "    \n",
    "    infl_novel = ''\n",
    "    # Plot novel SNPs\n",
    "    if plot_novel == True:\n",
    "        if known_SNPs_filename != '':\n",
    "            # Check if known SNPs file is empty\n",
    "            if __read_file(known_SNPs_filename) is None:\n",
    "                print('Empty file of novel SNPs')\n",
    "                pass # If file is empty or only has titles, only plot original SNPs\n",
    "            else:\n",
    "                # Plot novel SNPs, then same steps as before\n",
    "                to_be_removed_SNPs = __known_SNPs(known_SNPs_filename, ldPop, r2_threshold, known_SNPs_column_title)\n",
    "                novel_SNPs_df = __remove_known_SNPs(df, to_be_removed_SNPs, all_SNPs_column_title, novel_filename)\n",
    "\n",
    "                Pobsv_novel, logPobsv_novel, logPexpt_novel = __log_obsv_and_expt(novel_SNPs_df, p_value_column_title)\n",
    "                infl_novel = __get_inflation__(Pobsv_novel)\n",
    "                \n",
    "                ax.plot(logPexpt_novel, logPobsv_novel, linestyle='', marker='o', markersize=2, markeredgewidth=0.5,\n",
    "                        fillstyle='none', color='g', zorder=1)\n",
    "                \n",
    "                annotation = \"λ (novel) = \" + str(\"{0:.4f}\".format(infl_novel))\n",
    "                ax.annotate(annotation, xy=(0.7, 0.1), xycoords='axes fraction')\n",
    "        else: print('No file of known SNPs provided\\nOnly plot original SNPs')\n",
    "    \n",
    "\n",
    "    # Label x and y axis\n",
    "    ax.set_title(title)\n",
    "    ax.set_xlabel(xlabel)\n",
    "    ax.set_ylabel(ylabel)\n",
    "    annotation = \"λ = \" + str(\"{0:.4f}\".format(infl_original))\n",
    "    ax.annotate(annotation, xy=(0.7, 0.2), xycoords='axes fraction')\n",
    "    \n",
    "    fig.savefig(output)\n",
    "    \n",
    "    return (fig, ax, infl_original, infl_novel)  # Return for more custermizations\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_HDL.txt'\n",
    "df = __read_file(known_SNPs_filename)\n",
    "# df = pd.read_csv(known_SNPs_filename, delim_whitespace=True)\n",
    "\n",
    "# df = pd.read_csv(known_SNPs_filename, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DATE ADDED TO CATALOG</th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>FIRST AUTHOR</th>\n",
       "      <th>DATE</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>INITIAL SAMPLE SIZE</th>\n",
       "      <th>REPLICATION SAMPLE SIZE</th>\n",
       "      <th>...</th>\n",
       "      <th>PVALUE_MLOG</th>\n",
       "      <th>P-VALUE (TEXT)</th>\n",
       "      <th>OR or BETA</th>\n",
       "      <th>95% CI (TEXT)</th>\n",
       "      <th>PLATFORM [SNPS PASSING QC]</th>\n",
       "      <th>CNV</th>\n",
       "      <th>MAPPED_TRAIT</th>\n",
       "      <th>MAPPED_TRAIT_URI</th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <th>GENOTYPING TECHNOLOGY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6/21/17</td>\n",
       "      <td>28270201</td>\n",
       "      <td>Nagy R</td>\n",
       "      <td>3/7/17</td>\n",
       "      <td>Genome Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28270201</td>\n",
       "      <td>Exploration of haplotype research consortium i...</td>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>19,223 British ancestry individuals from 6863 ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>7.522879</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.867736</td>\n",
       "      <td>[0.56-1.18] mg/dl increase</td>\n",
       "      <td>Illumina [24111857] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST004207</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2/14/19</td>\n",
       "      <td>29507422</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>13.698970</td>\n",
       "      <td>(EA)</td>\n",
       "      <td>0.113000</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2/14/19</td>\n",
       "      <td>29507422</td>\n",
       "      <td>Hoffmann TJ</td>\n",
       "      <td>3/5/18</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/29507422</td>\n",
       "      <td>A large electronic-health-record-based genome-...</td>\n",
       "      <td>High density lipoprotein cholesterol levels</td>\n",
       "      <td>76,627 European ancestry individuals, 7,795 Hi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>13.698970</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.114000</td>\n",
       "      <td>unit increase</td>\n",
       "      <td>Affymetrix [at least 7091467] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST007140</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5/12/14</td>\n",
       "      <td>24097068</td>\n",
       "      <td>Willer CJ</td>\n",
       "      <td>10/6/13</td>\n",
       "      <td>Nat Genet</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/24097068</td>\n",
       "      <td>Discovery and refinement of loci associated wi...</td>\n",
       "      <td>HDL cholesterol</td>\n",
       "      <td>94,595 European ancestry individuals</td>\n",
       "      <td>93,982 European ancestry individuals</td>\n",
       "      <td>...</td>\n",
       "      <td>15.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.051000</td>\n",
       "      <td>[NR] unit decrease</td>\n",
       "      <td>NR [NR] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST002223</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6/26/17</td>\n",
       "      <td>28334899</td>\n",
       "      <td>Spracklen CN</td>\n",
       "      <td>2/21/17</td>\n",
       "      <td>Hum Mol Genet</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28334899</td>\n",
       "      <td>Association analyses of East Asian individuals...</td>\n",
       "      <td>HDL cholesterol levels</td>\n",
       "      <td>34,930 East Asian ancestry individuals, 187,16...</td>\n",
       "      <td>8,741 Chinese ancestry individuals</td>\n",
       "      <td>...</td>\n",
       "      <td>15.045757</td>\n",
       "      <td>(Trans-ethnic initial)</td>\n",
       "      <td>0.050600</td>\n",
       "      <td>[0.038-0.063] unit decrease (EA Beta values)</td>\n",
       "      <td>Affymetrix, Illumina [~ 1900000] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>high density lipoprotein cholesterol measurement</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004612</td>\n",
       "      <td>GCST004232</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG  PUBMEDID  FIRST AUTHOR     DATE        JOURNAL  \\\n",
       "0               6/21/17  28270201        Nagy R   3/7/17     Genome Med   \n",
       "1               2/14/19  29507422   Hoffmann TJ   3/5/18      Nat Genet   \n",
       "2               2/14/19  29507422   Hoffmann TJ   3/5/18      Nat Genet   \n",
       "3               5/12/14  24097068     Willer CJ  10/6/13      Nat Genet   \n",
       "4               6/26/17  28334899  Spracklen CN  2/21/17  Hum Mol Genet   \n",
       "\n",
       "                                   LINK  \\\n",
       "0  www.ncbi.nlm.nih.gov/pubmed/28270201   \n",
       "1  www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "2  www.ncbi.nlm.nih.gov/pubmed/29507422   \n",
       "3  www.ncbi.nlm.nih.gov/pubmed/24097068   \n",
       "4  www.ncbi.nlm.nih.gov/pubmed/28334899   \n",
       "\n",
       "                                               STUDY  \\\n",
       "0  Exploration of haplotype research consortium i...   \n",
       "1  A large electronic-health-record-based genome-...   \n",
       "2  A large electronic-health-record-based genome-...   \n",
       "3  Discovery and refinement of loci associated wi...   \n",
       "4  Association analyses of East Asian individuals...   \n",
       "\n",
       "                                 DISEASE/TRAIT  \\\n",
       "0                              HDL cholesterol   \n",
       "1  High density lipoprotein cholesterol levels   \n",
       "2  High density lipoprotein cholesterol levels   \n",
       "3                              HDL cholesterol   \n",
       "4                       HDL cholesterol levels   \n",
       "\n",
       "                                 INITIAL SAMPLE SIZE  \\\n",
       "0  19,223 British ancestry individuals from 6863 ...   \n",
       "1  76,627 European ancestry individuals, 7,795 Hi...   \n",
       "2  76,627 European ancestry individuals, 7,795 Hi...   \n",
       "3               94,595 European ancestry individuals   \n",
       "4  34,930 East Asian ancestry individuals, 187,16...   \n",
       "\n",
       "                REPLICATION SAMPLE SIZE  ... PVALUE_MLOG  \\\n",
       "0                                   NaN  ...    7.522879   \n",
       "1                                   NaN  ...   13.698970   \n",
       "2                                   NaN  ...   13.698970   \n",
       "3  93,982 European ancestry individuals  ...   15.000000   \n",
       "4    8,741 Chinese ancestry individuals  ...   15.045757   \n",
       "\n",
       "           P-VALUE (TEXT) OR or BETA  \\\n",
       "0                     NaN   0.867736   \n",
       "1                    (EA)   0.113000   \n",
       "2                     NaN   0.114000   \n",
       "3                     NaN   0.051000   \n",
       "4  (Trans-ethnic initial)   0.050600   \n",
       "\n",
       "                                  95% CI (TEXT)  \\\n",
       "0                    [0.56-1.18] mg/dl increase   \n",
       "1                                 unit increase   \n",
       "2                                 unit increase   \n",
       "3                            [NR] unit decrease   \n",
       "4  [0.038-0.063] unit decrease (EA Beta values)   \n",
       "\n",
       "                   PLATFORM [SNPS PASSING QC] CNV  \\\n",
       "0               Illumina [24111857] (imputed)   N   \n",
       "1     Affymetrix [at least 7091467] (imputed)   N   \n",
       "2     Affymetrix [at least 7091467] (imputed)   N   \n",
       "3                           NR [NR] (imputed)   N   \n",
       "4  Affymetrix, Illumina [~ 1900000] (imputed)   N   \n",
       "\n",
       "                                       MAPPED_TRAIT  \\\n",
       "0  high density lipoprotein cholesterol measurement   \n",
       "1  high density lipoprotein cholesterol measurement   \n",
       "2  high density lipoprotein cholesterol measurement   \n",
       "3  high density lipoprotein cholesterol measurement   \n",
       "4  high density lipoprotein cholesterol measurement   \n",
       "\n",
       "                       MAPPED_TRAIT_URI  STUDY ACCESSION  \\\n",
       "0  http://www.ebi.ac.uk/efo/EFO_0004612       GCST004207   \n",
       "1  http://www.ebi.ac.uk/efo/EFO_0004612       GCST007140   \n",
       "2  http://www.ebi.ac.uk/efo/EFO_0004612       GCST007140   \n",
       "3  http://www.ebi.ac.uk/efo/EFO_0004612       GCST002223   \n",
       "4  http://www.ebi.ac.uk/efo/EFO_0004612       GCST004232   \n",
       "\n",
       "          GENOTYPING TECHNOLOGY  \n",
       "0  Genome-wide genotyping array  \n",
       "1  Genome-wide genotyping array  \n",
       "2  Genome-wide genotyping array  \n",
       "3  Genome-wide genotyping array  \n",
       "4  Genome-wide genotyping array  \n",
       "\n",
       "[5 rows x 38 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "t2d_female_bmi_locuszoom = '/data100t1/share/DIAMANTE/mrmega_t2d_female_bmi_0619_rsids_locuszoom.result'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "known_SNPs_filename='/data100t1/share/DIAMANTE/mrmega_hdl_final_0619_rsids_locuszoom.result'\n",
    "# df = __read_file(known_SNPs_filename)\n",
    "\n",
    "# Good\n",
    "# df = pd.read_csv(known_SNPs_filename, sep='\\t')\n",
    "\n",
    "# df = pd.read_csv(known_SNPs_filename, delim_whitespace=True)\n",
    "df = pd.read_csv(known_SNPs_filename, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tg_locuszoom = '/data100t1/share/DIAMANTE/mrmega_tg_final_0619_rsids_locuszoom.result'\n",
    "tg = '/data100t1/share/DIAMANTE/mrmega_tg_final_0619_rsids.result'\n",
    "qqplot(filename=tg_locuszoom,output='Results/mrmega_tg_final_0619_rsids_locuszoom.png',p_value_column_title = 'P-value_association',title='mrmega_tg_final_0619_rsids_locuszoom')\n",
    "# qp.qqplot(filename=tg,output='Results/mrmega_tg_final_0619_rsids.png',p_value_column_title = 'P-value_association',title='mrmega_tg_final_0619_rsids')\n",
    "# qqplot(filename=tg_locuszoom,output='Results/mrmega_tg_final_0619_rsids_locuszoom_novel.png',\n",
    "#        p_value_column_title = 'P-value_association',\n",
    "#        title='mrmega_tg_final_0619_rsids_locuszoom_novel',\n",
    "#        plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_TG.txt',\n",
    "#        novel_filename = 'Results/mrmega_tg_final_0619_rsids_locuszoom_novel.csv',\n",
    "#        known_SNPs_column_title = 'SNPS',\n",
    "#        all_SNPs_column_title = 'MarkerName')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "t2d_female_locuszoom = '/data100t1/share/DIAMANTE/mrmega_t2d_female_0619_rsids_locuszoom.result'\n",
    "qqplot(filename=t2d_female_locuszoom,output='Results/mrmega_t2d_female_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_t2d_female_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_T2D.txt',\n",
    "       novel_filename = 'mrmega_t2d_female_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "t2d_female_bmi_locuszoom = '/data100t1/share/DIAMANTE/mrmega_t2d_female_bmi_0619_rsids_locuszoom.result'\n",
    "t2d_female_bmi = '/data100t1/share/DIAMANTE/mrmega_t2d_female_bmi_0619_rsids.result'\n",
    "# qp.qqplot(filename=t2d_female_bmi_locuszoom,output='Results/mrmega_t2d_female_bmi_0619_rsids_locuszoom.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_female_bmi_0619_rsids_locuszoom')\n",
    "# qp.qqplot(filename=t2d_female_bmi,output='Results/mrmega_t2d_female_bmi_0619_rsids.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_female_bmi_0619_rsids')\n",
    "\n",
    "# Plot novel\n",
    "qqplot(filename=t2d_female_bmi_locuszoom,output='Results/mrmega_t2d_female_bmi_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_t2d_female_bmi_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_T2D.txt',\n",
    "       novel_filename = 'mrmega_t2d_female_bmi_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n",
    "# ---------------\n",
    "t2d_locuszoom = '/data100t1/share/DIAMANTE/mrmega_t2d_final_0619_rsids_locuszoom.result'\n",
    "t2d = '/data100t1/share/DIAMANTE/mrmega_t2d_final_0619_rsids.result'\n",
    "# qp.qqplot(filename=t2d_locuszoom,output='Results/mrmega_t2d_final_0619_rsids_locuszoom.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_final_0619_rsids_locuszoom')\n",
    "# qp.qqplot(filename=t2d,output='Results/mrmega_t2d_final_0619_rsids.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_final_0619_rsids')\n",
    "\n",
    "qqplot(filename=t2d_locuszoom,output='Results/mrmega_t2d_final_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_t2d_final_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_T2D.txt',\n",
    "       novel_filename = 'mrmega_t2d_final_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n",
    "# ---------------\n",
    "t2d_final_bmi_locuszoom = '/data100t1/share/DIAMANTE/mrmega_t2d_final_bmi_0619_rsids_locuszoom.result'\n",
    "t2d_final_bmi = '/data100t1/share/DIAMANTE/mrmega_t2d_final_bmi_0619_rsids.result'\n",
    "# qp.qqplot(filename=t2d_final_bmi_locuszoom,output='Results/mrmega_t2d_final_bmi_0619_rsids_locuszoom.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_final_bmi_0619_rsids_locuszoom')\n",
    "# qp.qqplot(filename=t2d_final_bmi,output='Results/mrmega_t2d_final_bmi_0619_rsids.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_final_bmi_0619_rsids')\n",
    "\n",
    "qqplot(filename=t2d_final_bmi_locuszoom,output='Results/mrmega_t2d_final_bmi_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_t2d_final_bmi_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_T2D.txt',\n",
    "       novel_filename = 'mrmega_t2d_final_bmi_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n",
    "\n",
    "# ---------------\n",
    "t2d_male_locuszoom = '/data100t1/share/DIAMANTE/mrmega_t2d_male_0619_rsids_locuszoom.result'\n",
    "t2d_male = '/data100t1/share/DIAMANTE/mrmega_t2d_male_0619_rsids.result'\n",
    "# qp.qqplot(filename=t2d_male_locuszoom,output='Results/mrmega_t2d_male_0619_rsids_locuszoom.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_male_0619_rsids_locuszoom')\n",
    "# qp.qqplot(filename=t2d_male,output='Results/mrmega_t2d_male_0619_rsids.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_male_0619_rsids')\n",
    "\n",
    "qqplot(filename=t2d_male_locuszoom,output='Results/mrmega_t2d_male_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_t2d_male_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_T2D.txt',\n",
    "       novel_filename = 'mrmega_t2d_male_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n",
    "# ---------------\n",
    "t2d_male_bmi_locuszoom = '/data100t1/share/DIAMANTE/mrmega_t2d_male_bmi_0619_rsids_locuszoom.result'\n",
    "t2d_male_bmi = '/data100t1/share/DIAMANTE/mrmega_t2d_male_bmi_0619_rsids.result'\n",
    "# qp.qqplot(filename=t2d_male_bmi_locuszoom,output='Results/mrmega_t2d_male_bmi_0619_rsids_locuszoom.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_male_bmi_0619_rsids_locuszoom')\n",
    "# qp.qqplot(filename=t2d_male_bmi,output='Results/mrmega_t2d_male_bmi_0619_rsids.png',p_value_column_title = 'P-value_association',title='mrmega_t2d_male_bmi_0619_rsids')\n",
    "qqplot(filename=t2d_male_bmi_locuszoom,output='Results/mrmega_t2d_male_bmi_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_t2d_male_bmi_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_T2D.txt',\n",
    "       novel_filename = 'Results/mrmega_t2d_male_bmi_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n",
    "# ---------------\n",
    "tc_locuszoom = '/data100t1/share/DIAMANTE/mrmega_tc_final_0619_rsids_locuszoom.result'\n",
    "tc = '/data100t1/share/DIAMANTE/mrmega_tc_final_0619_rsids.result'\n",
    "# qp.qqplot(filename=tc_locuszoom,output='Results/mrmega_tc_final_0619_rsids_locuszoom.png',p_value_column_title = 'P-value_association',title='mrmega_tc_final_0619_rsids_locuszoom')\n",
    "# qp.qqplot(filename=tc,output='Results/mrmega_tc_final_0619_rsids.png',p_value_column_title = 'P-value_association',title='mrmega_tc_final_0619_rsids')\n",
    "qqplot(filename=tc_locuszoom,output='Results/mrmega_tc_final_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_tc_final_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_TC.txt',\n",
    "       novel_filename = 'Results/mrmega_tc_final_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n",
    "# ---------------\n",
    "tg_locuszoom = '/data100t1/share/DIAMANTE/mrmega_tg_final_0619_rsids_locuszoom.result'\n",
    "tg = '/data100t1/share/DIAMANTE/mrmega_tg_final_0619_rsids.result'\n",
    "# qp.qqplot(filename=tg_locuszoom,output='Results/mrmega_tg_final_0619_rsids_locuszoom.png',p_value_column_title = 'P-value_association',title='mrmega_tg_final_0619_rsids_locuszoom')\n",
    "# qp.qqplot(filename=tg,output='Results/mrmega_tg_final_0619_rsids.png',p_value_column_title = 'P-value_association',title='mrmega_tg_final_0619_rsids')\n",
    "qqplot(filename=tg_locuszoom,output='Results/mrmega_tg_final_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_tg_final_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_TG.txt',\n",
    "       novel_filename = 'Results/mrmega_tg_final_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n",
    "\n",
    "\n",
    "# ======================\n",
    "\n",
    "hdl_locuszoom = '/data100t1/share/DIAMANTE/mrmega_hdl_final_0619_rsids_locuszoom.result'\n",
    "qqplot(filename=hdl_locuszoom,output='Results/mrmega_hdl_final_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_hdl_final_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_HDL.txt',\n",
    "       novel_filename = 'Results/mrmega_hdl_final_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n",
    "ldl_locuszoom = '/data100t1/share/DIAMANTE/mrmega_ldl_final_0619_rsids_locuszoom.result'\n",
    "qqplot(filename=ldl_locuszoom,output='Results/mrmega_ldl_final_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_ldl_final_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_LDL.txt',\n",
    "       novel_filename = 'Results/mrmega_ldl_final_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n",
    "t2d_female_locuszoom = '/data100t1/share/DIAMANTE/mrmega_t2d_female_0619_rsids_locuszoom.result'\n",
    "qqplot(filename=t2d_female_locuszoom,output='Results/mrmega_t2d_female_0619_rsids_locuszoom_novel.png',\n",
    "       p_value_column_title = 'P-value_association',\n",
    "       title='mrmega_t2d_female_0619_rsids_locuszoom_novel',\n",
    "       plot_novel = True, known_SNPs_filename='/data100t1/share/DIAMANTE/GWAS_catalog_results_T2D.txt',\n",
    "       novel_filename = 'Results/mrmega_t2d_female_0619_rsids_locuszoom_novel.csv',\n",
    "       known_SNPs_column_title = 'SNPS',\n",
    "       all_SNPs_column_title = 'MarkerName')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
